---
title: "RNAseq ASCA Analysis"
author: "Ariana S Huffmyer"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

Analysis of RNAseq data using ASCA (ANOVA Simultaneous Component Analysis) to examine how gene expression changes across temperatures for each parental phenotype.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# The following setting is important, do not omit.
options(stringsAsFactors = FALSE) #Set Strings to character
```

# Load libraries

```{r}
library("tidyverse")
library("genefilter")
library("DESeq2")
library("RColorBrewer")
library("ALASCA")
```

# Data input and filtering

Import the data files.

```{r}
#load metadata sheet with sample name and developmental stage information
metadata <- read.csv("data/rna_seq/sample_rnaseq_metadata.csv", header = TRUE, sep = ",")%>%
  dplyr::select(sample, temperature, parent, symbiont, phenotype)
metadata$code<-paste0(metadata$temperature, "_", metadata$parent)
head(metadata)

#load gene count matrix generated from cluster computation
gcount <- as.data.frame(read.csv("data/rna_seq/Mcapitata2023_gene_count_matrix_STAR.csv", row.names="gene_id"), colClasses = double)
head(gcount)
```

Check that there are no genes with 0 counts across all samples.

```{r}
dim(gcount) 

gcount<-gcount %>%
     mutate(Total = rowSums(.[, 1:54]))%>%
    filter(!Total==0)%>%
    dplyr::select(!Total)

dim(gcount)
```

Conduct data filtering using pOverA approach.

```{r}
filt <- filterfun(pOverA(0.1,10))

#create filter for the counts data
gfilt <- genefilter(gcount, filt)

#identify genes to keep by count filter
gkeep <- gcount[gfilt,]

#identify gene lists
gn.keep <- rownames(gkeep)

#gene count data filtered in PoverA, P percent of the samples have counts over A
gcount_filt <- as.data.frame(gcount[which(rownames(gcount) %in% gn.keep),])

#How many rows do we have before and after filtering?
nrow(gcount) #Before
nrow(gcount_filt) #After
```

Prepare sample names.

```{r}
# Function to extract 'R' followed by numbers from column names
extract_R_numbers <- function(col_names) {
  sub("R(\\d+).*", "\\1", col_names)
}

# Apply the function to all column names
new_col_names <- sapply(names(gcount_filt), extract_R_numbers)

# Rename the columns
names(gcount_filt) <- paste0("R", new_col_names)

# remove "trim." from the name
names(gcount_filt) <- gsub("trim\\.", "", names(gcount_filt))

# Output the modified dataframe
names(gcount_filt)
length(names(gcount_filt))
```

# Format metadata and count data

Convert temperature and metadata categories to factors.

```{r}
metadata$temperature<-as.factor(metadata$temperature)
metadata$parent<-as.factor(metadata$parent)
metadata$symbiont<-as.factor(metadata$symbiont)
metadata$code<-as.factor(metadata$code)
```

Set levels of factors.

```{r}
metadata$temperature<-factor(metadata$temperature, levels=c("27", "30", "33"))
metadata$parent<-factor(metadata$parent, levels=c("Wildtype", "Bleached", "Nonbleached"))
metadata$code<-factor(metadata$code, levels=c("27_Wildtype", "27_Bleached", "27_Nonbleached", "30_Wildtype", "30_Bleached", "30_Nonbleached", "33_Wildtype", "33_Bleached", "33_Nonbleached"))
```

Reorder metadata to match gene count columns.

```{r}
list<-colnames(gcount_filt)
list<-as.factor(list)

metadata$sample<-as.factor(metadata$sample)

# Re-order the levels
metadata$sample <- factor(as.character(metadata$sample), levels=list)
# Re-order the data.frame
metadata_ordered <- metadata[order(metadata$sample),]
metadata_ordered$sample

head(metadata_ordered)

metadata_ordered$sample
colnames(gcount_filt)
head(gcount_filt)
```

# Normalize data using VST transformation

Create a DESeqDataSet and apply variance stabilizing transformation for normalization.

```{r}
#Set DESeq2 design
gdds <- DESeqDataSetFromMatrix(countData = gcount_filt,
                              colData = metadata_ordered,
                             design = ~temperature + parent + temperature:parent)
```

Calculate size factors and check if VST can be used (size factors should be <4).

```{r}
SF.gdds <- estimateSizeFactors(gdds)
print(sizeFactors(SF.gdds))

all(sizeFactors(SF.gdds)) < 4
```

Apply VST transformation.

```{r}
gvst <- vst(gdds, blind=TRUE)
head(assay(gvst), 3)
dim(gvst)
```

Extract normalized counts.

```{r}
normalized_counts <- assay(gvst)
dim(normalized_counts)
head(normalized_counts)
```

# Prepare data for ASCA analysis

Create long format data with sample, temperature, parent, gene, and value columns.

```{r}
# Transpose and convert to dataframe
normalized_df <- as.data.frame(t(normalized_counts))
normalized_df$sample <- rownames(normalized_df)
rownames(normalized_df) <- NULL

# Add metadata
normalized_df <- normalized_df %>%
  left_join(metadata_ordered %>% dplyr::select(sample, temperature, parent), by = "sample")

# Reorder columns to put metadata first
normalized_df <- normalized_df %>%
  dplyr::select(sample, temperature, parent, everything())

head(normalized_df[, 1:10])
```

Convert to long format for ALASCA.

```{r}
long_data <- normalized_df %>%
  pivot_longer(cols = -c(sample, temperature, parent), 
               names_to = "variable", 
               values_to = "value")

str(long_data)
head(long_data)
```

View any NA or Inf values. 

```{r}
long_data%>%
  filter(if_any(everything(), is.na))

long_data%>%
  filter(is.infinite(value))
```

No NA or INF values. 

# Run ASCA analysis for all parental phenotypes

Start with 10 runs, increase to 100 for full analysis. 

Run ASCA model with temperature and phenotype as the main effect and sample as random effect.

```{r}
res_all <- ALASCA(
  long_data,
  value ~ temperature * parent + (1|sample),
  n_validation_runs = 10,
  validate = TRUE,
  reduce_dimensions = TRUE
)
```

Generate scree plot.

```{r}
pdf(file="figures/rna_seq/asca/scree_plot.pdf", width=6, height=4)
plot(res_all, component = seq(1,10), effect = 1, type = 'scree')
dev.off()
```

Plot significant components (>1% variance explained).

```{r}
# PC1
pdf(file="figures/rna_seq/asca/PC1.pdf", width=10, height=4)
plot(res_all, effect = 1, component = 1, type = 'effect', 
     n_limit=10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis=TRUE,
     my_theme = theme_classic())
dev.off()

# PC1
pdf(file="figures/rna_seq/asca/PC2.pdf", width=10, height=4)
plot(res_all, effect = 1, component = 2, type = 'effect', 
     n_limit=10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis=TRUE,
     my_theme = theme_classic())
dev.off()

# PC3
pdf(file="figures/rna_seq/asca/PC3.pdf", width=10, height=4)
plot(res_all, effect = 1, component = 3, type = 'effect', 
     n_limit=10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis=TRUE,
     my_theme = theme_classic())
dev.off()

pdf(file="figures/rna_seq/asca/PC4.pdf", width=10, height=4)
plot(res_all, effect = 1, component = 4, type = 'effect', 
     n_limit=10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis=TRUE,
     my_theme = theme_classic())
dev.off()

pdf(file="figures/rna_seq/asca/PC5.pdf", width=10, height=4)
plot(res_all, effect = 1, component = 5, type = 'effect', 
     n_limit=10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis=TRUE,
     my_theme = theme_classic())
dev.off()

pdf(file="figures/rna_seq/asca/PC6.pdf", width=10, height=4)
plot(res_all, effect = 1, component = 6, type = 'effect', 
     n_limit=10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis=TRUE,
     my_theme = theme_classic())
dev.off()

pdf(file="figures/rna_seq/asca/PC7.pdf", width=10, height=4)
plot(res_all, effect = 1, component = 7, type = 'effect', 
     n_limit=10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis=TRUE,
     my_theme = theme_classic())
dev.off()

pdf(file="figures/rna_seq/asca/PC8.pdf", width=10, height=4)
plot(res_all, effect = 1, component = 8, type = 'effect', 
     n_limit=10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis=TRUE,
     my_theme = theme_classic())
dev.off()

pdf(file="figures/rna_seq/asca/PC9.pdf", width=10, height=4)
plot(res_all, effect = 1, component = 9, type = 'effect', 
     n_limit=10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis=TRUE,
     my_theme = theme_classic())
dev.off()

pdf(file="figures/rna_seq/asca/PC10.pdf", width=10, height=4)
plot(res_all, effect = 1, component = 10, type = 'effect', 
     n_limit=10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis=TRUE,
     my_theme = theme_classic())
dev.off()
```

Plot top genes contributing to each component.

```{r}
# PC1 genes
pdf(file="figures/rna_seq/asca/PC1_genes.pdf", width=12, height=6)
plot(res_all, component = 1, effect = 1, type = 'prediction', n_limit = 10, variable = c())
dev.off()

# PC2 genes
pdf(file="figures/rna_seq/asca/PC2_genes.pdf", width=12, height=6)
plot(res_all, component = 2, effect = 1, type = 'prediction', n_limit = 10, variable = c())
dev.off()

# PC3 genes
pdf(file="figures/rna_seq/asca/PC3_genes.pdf", width=12, height=6)
plot(res_all, component = 3, effect = 1, type = 'prediction', n_limit = 10, variable = c())
dev.off()

# PC4 genes
pdf(file="figures/rna_seq/asca/PC4_genes.pdf", width=12, height=6)
plot(res_all, component = 4, effect = 1, type = 'prediction', n_limit = 10, variable = c())
dev.off()

# PC5 genes
pdf(file="figures/rna_seq/asca/PC5_genes.pdf", width=12, height=6)
plot(res_all, component = 5, effect = 1, type = 'prediction', n_limit = 10, variable = c())
dev.off()

# PC6 genes
pdf(file="figures/rna_seq/asca/PC6_genes.pdf", width=12, height=6)
plot(res_all, component = 6, effect = 1, type = 'prediction', n_limit = 10, variable = c())
dev.off()

# PC7 genes
pdf(file="figures/rna_seq/asca/PC7_genes.pdf", width=12, height=6)
plot(res_all, component = 7, effect = 1, type = 'prediction', n_limit = 10, variable = c())
dev.off()

# PC8 genes
pdf(file="figures/rna_seq/asca/PC8_genes.pdf", width=12, height=6)
plot(res_all, component = 8, effect = 1, type = 'prediction', n_limit = 10, variable = c())
dev.off()

# PC9 genes
pdf(file="figures/rna_seq/asca/PC9_genes.pdf", width=12, height=6)
plot(res_all, component = 9, effect = 1, type = 'prediction', n_limit = 10, variable = c())
dev.off()

# PC10 genes
pdf(file="figures/rna_seq/asca/PC10_genes.pdf", width=12, height=6)
plot(res_all, component = 10, effect = 1, type = 'prediction', n_limit = 10, variable = c())
dev.off()
```

Extract top genes from each component.

```{r}
loadings1 <- get_loadings(res_all, component=1, effect=1, n_limit=20)
list1 <- loadings1[[1]]$covars

loadings2 <- get_loadings(res_all, component=2, effect=1, n_limit=20)
list2 <- loadings2[[1]]$covars

loadings3 <- get_loadings(res_all, component=3, effect=1, n_limit=20)
list3 <- loadings3[[1]]$covars

loadings4 <- get_loadings(res_all, component=4, effect=1, n_limit=20)
list4 <- loadings4[[1]]$covars

loadings5 <- get_loadings(res_all, component=5, effect=1, n_limit=20)
list5 <- loadings5[[1]]$covars

loadings6 <- get_loadings(res_all, component=6, effect=1, n_limit=20)
list6 <- loadings6[[1]]$covars

loadings7 <- get_loadings(res_all, component=7, effect=1, n_limit=20)
list7 <- loadings7[[1]]$covars

loadings8 <- get_loadings(res_all, component=8, effect=1, n_limit=20)
list8 <- loadings8[[1]]$covars

loadings9 <- get_loadings(res_all, component=9, effect=1, n_limit=20)
list9 <- loadings9[[1]]$covars

loadings10 <- get_loadings(res_all, component=10, effect=1, n_limit=20)
list10 <- loadings10[[1]]$covars

# Combine all lists
list_wt <- data.frame(
  PC1 = c(list1, rep(NA, max(0, 20 - length(list1)))),
  PC2 = c(list2, rep(NA, max(0, 20 - length(list2)))),
  PC3 = c(list3, rep(NA, max(0, 20 - length(list3)))),
  PC4 = c(list4, rep(NA, max(0, 20 - length(list4)))),
  PC5 = c(list5, rep(NA, max(0, 20 - length(list5)))),
  PC6 = c(list6, rep(NA, max(0, 20 - length(list6)))),
  PC7 = c(list7, rep(NA, max(0, 20 - length(list7)))),
  PC8 = c(list8, rep(NA, max(0, 20 - length(list8)))),
  PC9 = c(list9, rep(NA, max(0, 20 - length(list9)))),
  PC10 = c(list10, rep(NA, max(0, 20 - length(list10))))
)

write.csv(list_wt, file="output/rna_seq/asca/PC1-10_top_genes.csv", row.names=FALSE)
```
