---
title: "RNAseq DEG Functional Enrichment Analysis"
author: "Ariana S Huffmyer"
date: "4/21/2024"
output: html_document
editor_options: 
  chunk_output_type: console
---

Functional enrichment of M. capitata larvae DEGs.  

# Set up 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# The following setting is important, do not omit.
options(stringsAsFactors = FALSE) #Set Strings to character
```

```{r}
if ("tidyverse" %in% rownames(installed.packages()) == 'FALSE') install.packages('tidyverse') 
if ("genefilter" %in% rownames(installed.packages()) == 'FALSE') install.packages('genefilter') 
if ("DESeq2" %in% rownames(installed.packages()) == 'FALSE') install.packages('DESeq2') 
if ("RColorBrewer" %in% rownames(installed.packages()) == 'FALSE') install.packages('RColorBrewer') 
if ("WGCNA" %in% rownames(installed.packages()) == 'FALSE') install.packages('WGCNA') 
if ("flashClust" %in% rownames(installed.packages()) == 'FALSE') install.packages('flashClust') 
if ("gridExtra" %in% rownames(installed.packages()) == 'FALSE') install.packages('gridExtra') 
if ("ComplexHeatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('ComplexHeatmap') 
if ("goseq" %in% rownames(installed.packages()) == 'FALSE') install.packages('goseq') 
if ("dplyr" %in% rownames(installed.packages()) == 'FALSE') install.packages('dplyr') 
if ("clusterProfiler" %in% rownames(installed.packages()) == 'FALSE') install.packages('clusterProfiler') 
if ("pheatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('pheatmap') 
if ("magrittr" %in% rownames(installed.packages()) == 'FALSE') install.packages('magrittr') 
if ("rtracklayer" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rtracklayer")
if ("GenomicRanges" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GenomicRanges")
if ("plyranges" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("plyranges")
if ("GSEABase" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GSEABase")
if ("GOSim" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GOSim")
if ("stats" %in% rownames(installed.packages()) == 'FALSE') install.packages("stats")
if ("ggdendro" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("ggdendro")
if ("GO.db" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GO.db")
if ("rrvgo" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rrvgo")

#BiocManager::install("org.Ce.eg.db", force=TRUE) #install if needed 

library("tidyverse")
library("genefilter")
library("DESeq2")
library("RColorBrewer")
library("WGCNA")
library("flashClust")
library("gridExtra")
library("ComplexHeatmap")
library("goseq")
library("dplyr")
library("clusterProfiler")
library("pheatmap")
library("magrittr")
library("rtracklayer")
library("GenomicRanges")
library("plyranges")
library("GSEABase")
library("GOSim")
library("stats")
library("ggdendro")
library("GO.db")
library("rrvgo")
```

# Read in data files 

1. Read in file with vst transformed counts of all genes detected and kept after filtering. 
```{r}
all_genes<-read.csv("output/rna_seq/all_genes.csv")
```

2. Read in file of DEG's between parental phenotypes
```{r}
temperature_DEG_list<-read.csv("output/rna_seq/temperature_deg_list.csv")
temperature_DEG_counts<-read.csv("output/rna_seq/temperature_deg_counts.csv")

temperature_DEG_list$contrast<-factor(temperature_DEG_list$contrast, levels=c("30vs27", "33vs27", "33vs30"))
```

3. Read in file of DEG's between temperatures 
```{r}
parent_DEG_list<-read.csv("output/rna_seq/parent_deg_list.csv")
parent_DEG_counts<-read.csv("output/rna_seq/parent_deg_counts.csv")
```

4. Read in file of DEG's between NB and B at 30-33°C 
```{r}
interaction_DEG_list<-read.csv("output/rna_seq/NBvsB_30-33_deg_list.csv")
interaction_DEG_counts<-read.csv("output/rna_seq/NBvsB_30-33_deg_counts.csv")
```

The list dataframes include DESeq2 results for each gene identified as a DEG in each contrast.  

The counts dataframes include gene counts for each gene in the DEG list for each sample. The number of genes may be lower than the number in the full list, because some genes are shared between contrasts in the list.  

# Read in annotation files 

Downloaded functional annotation (version 3) from: http://cyanophora.rutgers.edu/montipora/Montipora_capitata_HIv3.genes.EggNog_results.txt.gz (EggNog) and http://cyanophora.rutgers.edu/montipora/Montipora_capitata_HIv3.genes.KEGG_results.txt.gz (KEGG). 

Scripts to generate this file are from Mcapitata 2020 project scripts here: https://github.com/AHuffmyer/EarlyLifeHistory_Energetics 

Downloaded on 21 April 2024. Unzipped on desktop. I then removed the # in #query in the first column, otherwise it does not read in column names.    

```{r}
Mcap.annot <- read.table("data/rna_seq/Montipora_capitata_HIv3.genes.EggNog_results.txt",  quote="", sep="\t", header=TRUE)
dim(Mcap.annot)

# column names
head(Mcap.annot)
tail(Mcap.annot)

Mcap.annot<-Mcap.annot%>%
  rename(gene=query)
```

This functional annotation as 24,072 genes. This is 44% of total protein predicted genes described in the publication associated with this annotation. https://academic.oup.com/gigascience/article/doi/10.1093/gigascience/giac098/6815755 

We had 24,005 genes in our dataset that were detected and present at counts >10. 

Match up genes in gene list file to annotation file
```{r}
names(Mcap.annot)

#remove sample column name 
probes = names(all_genes)
#probes <- gsub("sample", "", probes)
#probes <- probes[probes != ""]

probes2annot = match(probes, Mcap.annot$gene) #match genes in datExpr to genes in annotation file, note I removed the # before query in this file before loading

# The following is the number of probes without annotation 
sum(is.na(probes2annot))

row_nas<-which(is.na(probes2annot))

#view the genes that do not have a match in the annotation file
missing<-as.data.frame(probes[row_nas])
print(missing)
```

There are 9390 genes from our data set that are not in the annotation file but that are in the genome. We will need to BLAST these to obtain functional information.  

## Reduce annotation file to only contain genes detected in our dataset.  

```{r}
filtered_Mcap.annot <- Mcap.annot[Mcap.annot$gene %in% probes, ]
dim(filtered_Mcap.annot)
```

The annotation file now only contains genes that were detected in our dataset that have annotation information.  

## Generate a vector of gene lenghts 

```{r}
#import file
gff <- rtracklayer::import("data/rna_seq/Montipora_capitata_HIv3.genes_fixed.gff3") #if this doesn't work, restart R and try again 

transcripts <- subset(gff, type == "transcript") #keep only transcripts 

transcripts_gr <- makeGRangesFromDataFrame(transcripts, keep.extra.columns=TRUE) #extract length information

transcript_lengths <- width(transcripts_gr) #isolate length of each transcript

seqnames<-transcripts_gr$ID #extract list of gene id 

lengths<-cbind(seqnames, transcript_lengths)

lengths<-as.data.frame(lengths) #convert to data frame
```

Add in length to the annotation file.    

```{r}
filtered_Mcap.annot$length<-lengths$transcript_lengths[match(filtered_Mcap.annot$gene, lengths$seqnames)]

which(is.na(filtered_Mcap.annot$length)) #all genes have lengths 
```

Format GO terms to remove dashes and quotes.  
```{r}
#remove dashes, remove quotes from columns and separate by semicolons (replace , with ;) in KEGG_ko, KEGG_pathway, and Annotation.GO.ID columns
head(filtered_Mcap.annot$GOs)
filtered_Mcap.annot$GOs <- gsub(",", ";", filtered_Mcap.annot$GOs)
filtered_Mcap.annot$GOs <- gsub('"', "", filtered_Mcap.annot$GOs)
filtered_Mcap.annot$GOs <- gsub("-", NA, filtered_Mcap.annot$GOs)
head(filtered_Mcap.annot$GOs)
```

# Goseq settings and information  

**nullp function** 

nullp: Calculate probability weighting function

It is essential that the entire analysis pipeline, from summarizing raw reads through to using goseq be done in just one gene identifier format. If your data is in a different format you will need to obtain the gene lengths and supply them to the nullp function using the bias.data arguement. Converting to a supported format from another format should be avoided whenever possible as this will almost always result in data loss.

NAs are allowed in the bias.data vector if you do not have information about a certain gene. Setting a gene to NA is preferable to removing it from the analysis.

gene.vector: A named binary vector where 1 represents DE, 0 not DE and the names are gene IDs. List of all genes in reference list (either all genes detected or all genes in genome).  

genome: A string identifying the genome that genes refer to if using a model organism. We don't use this because we don't have a model organism.  

ID.vector: 	A string identifying the gene identifier used by genes. This is our list of DEGs.  

bias.data: A numeric vector containing the data on which the DE may depend. Usually this is the median transcript length of each gene in bp. If set to NULL nullp will attempt to fetch length using getlength. This is our gene length vector LENGTH.vector. 


pwf: The pwf argument is almost always the output of the function nullp. This is a data frame with 3 columns, named "DEgenes", "bias.data" and "pwf" with the rownames set to the gene names. Each row corresponds to a gene with the DEgenes column specifying if the gene is DE (1 for DE, 0 for not DE), the bias.data column giving the numeric value of the DE bias being accounted for (usually the gene length or number of counts) and the pwf column giving the genes value on the probability weighting function. This includes a list of full genes generated from our ALL.vector, which here is the list of all genes present in our dataset. Alternatively, this can be a list of all genes in the genome. 

Our list of background genes is all genes expressed with a count of at least 10 in at least 11% of samples (pOverA filtering).  

*Discussion point: Should you use genes from your dataset or genes from the genome? When would you use either option?*

ID.vector: list of DEGs that we are testing 

gene2cat: To use your own gene to category mapping with goseq, use the gene2cat arguement. This arguement takes a data.frame, with one column containing gene IDs and the other containing the associated categories. As the mapping from gene <-> category is in general many to many there will be multiple rows containing the same gene identifier. Alternatively, gene2cat can take a list, where the names are the genes and the entries are the GO categories associated with the genes. If gene2cat is left as NULL, goseq attempts to use getgo to fetch GO catgeory to gene identifier mappings.

Wallenius: "Wallenius" approximates the true distribution of numbers of members of a category amongst DE genes by the Wallenius non-central hypergeometric distribution. This distribution assumes that within a category all genes have the same probability of being chosen. Therefore, this approximation works best when the range in probabilities obtained by the probability weighting function is small. "Wallenius" is the recommended method for calculating p-values.

test.cats: Valid options for the test.cats arguement are any combination of "GO:CC", "GO:BP", "GO:MF" & "KEGG". The three GO terms refer to the Cellular Component, Biological Process and Molecular Function respectively. "KEGG" refers to KEGG pathways.

# rrvgo settings and information 

**calculateSimMatrix** 
https://rdrr.io/bioc/rrvgo/f/vignettes/rrvgo.Rmd  

Calculate the score similarity matrix between terms. The function calculateSimMatrix takes a list of GO terms for which the semantic simlarity is to be calculated, an OrgDb object for an organism, the ontology of interest and the method to calculate the similarity scores.

Example:  
calculateSimMatrix(go_results$GOterm, orgdb="org.Ce.eg.db", ont=c("BP"), method="Rel")

go_results$GOterm: vector of GO terms  

orgdb: reference database; one of org.* Bioconductor packages (the package name, or the package itself). Here we will use C. elegans.  

ont: ontlogy. One of c("BP", "MF", "CC")  

method: distance method. One of the supported methods by GOSemSim: c("Resnik", "Lin", "Rel", "Jiang", "Wang")

https://academic.oup.com/bioinformatics/article/26/7/976/213143
https://yulab-smu.top/biomedical-knowledge-mining-book/  

Four IC-based (Resnik's, Lin's, Jiang and Conrath's and Schlicker's) and one graph-based (Wang's) semantic similarity measure algorithms mentioned before are selected to be integrated in GOSemSim, and can be selected by setting the ‘method’ parameter of the package functions to ‘Resnik’, ‘Lin’, ‘Jiang’, ‘Rel’ and ‘Wang’, respectively. The Resnik's, Lin's and Jiang and Conrath's algorithms are the most common semantic similarity measures used with GO (Pesquita et al., 2009). Several assessment results had shown that the Resnik's measure had consistently high correlation with sequence similarity (Lord et al., 2003; Mistry and Pavlidis, 2008; Pesquita et al., 2008) and gene co-expression (Sevilla et al., 2005). By using a best-match average combination strategy, Pesquita et al. found that Jiang and Conrath's measure had the highest correlation with sequence similarity (Pesquita et al., 2009). The Schlicker's measure had been found to perform better than Resnik's measure in distinguishing orthologous gene products from gene products with other levels of sequence similarity (Schlicker et al., 2006). The Wang's measure had also been shown to produce more accurate results than Resnik's measure in clustering gene pairs according to their semantic similarity (Wang et al., 2007). The details about the semantic similarity measure algorithms used in GOSemSim can be found in the user's manual (Supplementary Material 1).

**reduceSimMatrix**  

rrvgo provides the reduceSimMatrix function for that. It takes as arguments i) the similarity matrix, ii) an optional named vector of scores associated to each GO term, iii) a similarity threshold used for grouping terms, and iv) an orgdb object.

reduceSimMatrix selects as the group representative the term with the higher score within the group. In case the vector of scores is not available, reduceSimMatrix will get the GO term size from the OrgDb object and use it as the score, thus favoring broader terms. Please note that scores are interpreted in the direction that higher are better, therefore if you use p-values as scores, minus log-transform them before. This is what we do in the script below.  

Higher thresholds force higher similarity between terms of a groups, resulting in more groups containing less similar terms.

scores: named vector with scores (weights) assigned to each term. Higher is better. Can be NULL (default, means no scores. In this case, a default score based on set size is assigned, thus favoring larger sets). Note: if you have p-values as scores, consider -1*log-transforming them ('-log(p)')
                              
threshold: similarity threshold (0-1). Some guidance: Large (allowed similarity=0.9), Medium (0.7), Small (0.5), Tiny (0.4) Defaults to Medium (0.7)     

simMatrix: a (square) similarity matrix generated from calculateSimMatrix  

Example:   
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm) 
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")

# Temperature DEGs 

## Add in annotation information

Merge in relevant annotation information to the DEG list file.  

```{r}
temperature_DEG_list<-left_join(temperature_DEG_list, filtered_Mcap.annot, by="gene")
```

## Plot fold change of temperature DEGs

```{r}
temp_deg_FC <- temperature_DEG_list %>% 
  dplyr::arrange(log2FoldChange) %>%
  
  ggplot(aes(x=log2FoldChange, y=reorder(gene, -log2FoldChange), colour=padj)) +
  facet_grid(~contrast, scales="free_y")+
  geom_point(size=2)+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  geom_segment(aes(x=0, xend=log2FoldChange, y=gene, yend=gene),linewidth=1) +
  geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("Gene")+
  theme(legend.position = "right",
        axis.text.y = element_blank(),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Fold Change: Temperature DEGs'); temp_deg_FC

ggsave(plot=temp_deg_FC, "figures/rna_seq/functional_enrichment/temperature_DEG_foldchange.png", height=12, width=12)
```

## Prepare up and down regulated dataframes 

```{r}
dim(temperature_DEG_list)

temperature_up_list<-temperature_DEG_list%>%
  filter(log2FoldChange>0)

dim(temperature_up_list)

temperature_down_list<-temperature_DEG_list%>%
  filter(log2FoldChange<0)

dim(temperature_down_list)
```
1182 total genes with 518 up regulated and 664 down regulated. We are keeping all contrasts in these lists and we will run a loop to conduct enrichment for each contrast.    

## Conduct enrichment for up regulated genes 

Make a list of contrasts to loop through.  

```{r}
levels(temperature_up_list$contrast)
temp_contrasts<-c("30vs27", "33vs27", "33vs30")
```

Conduct functional enrichment for each contrast using goseq and rrvgo. Compare against all genes detected in the dataset. I am choosing to compare against genes in our dataset because we are looking at one developmental stage (larvae). Therefore, I do not want to conduct enrichment of all genes in the genome because many functions including calcification and gametogenesis do not apply to the biological function of larvae.  

Conduct at biological process level. Filter for adjusted p-value <0.05.  
```{r}
for (temp in temp_contrasts) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(filtered_Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(filtered_Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- temperature_up_list%>%
      filter(contrast==temp)%>%
      pull(gene)
    
    ##Get a list of GO Terms for all genes detected in our dataset 
    GO.terms <- filtered_Mcap.annot%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("output/rna_seq/functional_enrichment/goseq_temperature_up_", temp, ".csv"))
    
}
```

Create an empty dataframe to populate (only do this once).  

```{r}
temp_go_results <- data.frame()
```

Run a loop to filter enrichment for each temperature contrast using rrvgo. There were no enriched terms for 27 vs 30°C. Proceed with only 27vs33 and 30vs33.    
```{r}
temp_contrasts<-c("33vs27", "33vs30")

for (temp in temp_contrasts) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("output/rna_seq/functional_enrichment/goseq_temperature_up_", temp, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      #filter(numInCat>5)%>%
      arrange(., bh_adjust)
    
    print(temp)
    print(length(unique(go_results$GOterm)))
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log10(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$direction<-c("upregulated")
    go_results$contrast<-temp
    
    print(length(unique(go_results$GOterm)))
    
    #add to data frame 
    temp_go_df <- go_results
    temp_go_results <- rbind(temp_go_results, temp_go_df) 

}
```
There are 175 terms before sem sim for 33 vs 27 and 115 after. There are 42 terms before sem sim for 33 vs 30 and 30 after.  

```{r}
length(unique(temp_go_results$GOterm))
```
126 total enriched GO terms that are upregulated.  

## Conduct enrichment for down regulated genes 

Make a list of contrasts to loop through. 27vs30 had no enriched terms.   

```{r}
temp_contrasts<-c("30vs27", "33vs27", "33vs30")
```

Conduct at biological process level. Filter for adjusted p-value <0.05.   
```{r}
for (temp in temp_contrasts) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(filtered_Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(filtered_Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- temperature_down_list%>%
      filter(contrast==temp)%>%
      pull(gene)
    
    ##Get a list of GO Terms for all genes expressed
    GO.terms <- filtered_Mcap.annot%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("output/rna_seq/functional_enrichment/goseq_temperature_down_", temp, ".csv"))
    
}
```

Run a loop to filter enrichment for each temperature contrast using rrvgo. There are no enriched terms for 30vs27 and it is not used in this next section.   
```{r}
temp_contrasts<-c("33vs27", "33vs30")

for (temp in temp_contrasts) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("output/rna_seq/functional_enrichment/goseq_temperature_down_", temp, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      #filter(numInCat>5)%>%
      arrange(., bh_adjust)
    
    print(temp)
    print(length(unique(go_results$GOterm)))
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log10(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$direction<-c("downregulated")
    go_results$contrast<-temp
    
    print(length(unique(go_results$GOterm)))
     
    #add to data frame 
    temp_go_df <- go_results
    temp_go_results <- rbind(temp_go_results, temp_go_df) 

}
```
There are 107 enriched downregulated terms at 33vs27 before sem sim and 100 after. There are 110 terms at 33vs30 before sem sim and 102 after. 

```{r}
length(unique(temp_go_results$GOterm))
```
There are now 245 enriched GO terms (126 were upregulated, leaving 119 downregulated). 

## Plot enrichment for up and down regulated genes for each contrast 

Upregulated genes 
```{r}
temp_go_plot_up <- temp_go_results %>% 
  filter(direction=="upregulated")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=contrast, y=term, fill=over_represented_pvalue, size=percHits, colour=over_represented_pvalue)) +
  facet_grid(ParentTerm ~ ., scales="free_y", labeller = label_wrap_gen(multi_line = TRUE))+
  geom_point()+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  scale_fill_gradientn(colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  scale_x_discrete(limits=c("30vs27", "33vs30", "33vs27"))+
  theme_classic() +
  ylab("GO Term")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Enrichment: Upregulated Temperature DEGs'); temp_go_plot_up

ggsave(plot=temp_go_plot_up, "figures/rna_seq/functional_enrichment/temperature_GO_upregulated.png", height=40, width=10, limitsize=FALSE)
```

Downregulated genes 
```{r}
temp_go_plot_down <- temp_go_results %>% 
  filter(direction=="downregulated")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=contrast, y=term, colour=over_represented_pvalue, size=perHits)) +
  facet_grid(ParentTerm ~ ., scales="free", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=percHits))+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  ylab("GO Term")+
  scale_x_discrete(limits=c("30vs27", "33vs30", "33vs27"))+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Enrichment: Downregulated Temperature DEGs'); temp_go_plot_down

ggsave(plot=temp_go_plot_down, "figures/rna_seq/functional_enrichment/temperature_GO_downregulated.png", height=40, width=10, limitsize=FALSE)
```

Plot with up and down regulated with facet by comparison. 

```{r}
temp_go_plot_parent_data <- temp_go_results %>% 
  group_by(ParentTerm, contrast, direction)%>%
  summarise(numberTerms=length(unique(GOterm)))%>%
  arrange(ParentTerm)

# First, let's find the unique ParentTerm values
unique_parent_terms <- unique(temp_go_plot_parent_data$ParentTerm)

# Determine the number of unique ParentTerm values
num_unique <- length(unique_parent_terms)

# Determine the midpoint for splitting into two batches
midpoint <- ceiling(num_unique / 2)

temp_go_plot_parent_data <- temp_go_plot_parent_data %>%
  mutate(batch = ifelse(match(ParentTerm, unique_parent_terms) <= midpoint, 1, 2))

temp_go_plot_parent_data$ParentTerm <- factor(temp_go_plot_parent_data$ParentTerm) 
temp_go_plot_parent_data$ParentTerm <- factor(temp_go_plot_parent_data$ParentTerm, levels = rev(levels(temp_go_plot_parent_data$ParentTerm)))

temp_go_plot_parentA<-temp_go_plot_parent_data%>%
  filter(batch=="1")%>%
  
  ggplot(aes(x=contrast, y=ParentTerm, colour=direction)) +
  facet_grid(~ direction, scales="free_y", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=numberTerms))+
  scale_size(name="Num. GO Terms", breaks=c(2,10,20,50,80), range = c(1, 5))+
  scale_colour_manual(values=c("blue", "red"))+
  scale_x_discrete(limits=c("30vs27", "33vs30", "33vs27"))+
  theme_classic() +
  xlab("")+
  ylab("Parent Term")+
  ggtitle('Temperature DEGs')+
  theme(legend.position = "none",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"),
        axis.text.x = element_text(size=12, angle=45, vjust=0.5))+
      panel_border(color="black"); temp_go_plot_parentA

temp_go_plot_parentB<-temp_go_plot_parent_data%>%
  filter(batch=="2")%>%
  
  ggplot(aes(x=contrast, y=ParentTerm, colour=direction)) +
  facet_grid(~ direction, scales="free_y", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=numberTerms))+
  scale_size(name="Num. GO Terms", breaks=c(2,10,20,50,80), range = c(1, 5))+
  scale_colour_manual(values=c("blue", "red"))+
  scale_x_discrete(limits=c("30vs27", "33vs30", "33vs27"))+
  theme_classic() +
    xlab("")+
  ylab("")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_blank(),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"),
        axis.text.x = element_text(size=12, angle=45, vjust=0.5))+
        panel_border(color="black")+
  ggtitle(''); temp_go_plot_parentB

temp_go_grid<-plot_grid(temp_go_plot_parentA, temp_go_plot_parentB, ncol=2, nrow=1, align="h", rel_widths=c(0.8,1))

ggsave(plot=temp_go_grid, "figures/rna_seq/functional_enrichment/temperature_GO_parent.png", height=6, width=12, limitsize=FALSE)
```

# Parent DEGs 

## Add in annotation information

Merge in relevant annotation information to the DEG list file.  

```{r}
parent_DEG_list<-left_join(parent_DEG_list, filtered_Mcap.annot, by="gene")
```

## Plot fold change of temperature DEGs

```{r}
parent_deg_FC <- parent_DEG_list %>% 
  dplyr::arrange(log2FoldChange) %>%
  
  ggplot(aes(x=log2FoldChange, y=reorder(gene, -log2FoldChange), colour=padj)) +
  facet_grid(~contrast, scales="free_y")+
  geom_point(size=2)+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  geom_segment(aes(x=0, xend=log2FoldChange, y=gene, yend=gene),linewidth=1) +
  geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("Gene")+
  theme(legend.position = "right",
        axis.text.y = element_blank(),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Fold Change: Parent DEGs'); parent_deg_FC

ggsave(plot=parent_deg_FC, "figures/rna_seq/functional_enrichment/parent_DEG_foldchange.png", height=12, width=12)
```

## Prepare up and down regulated dataframes 

```{r}
dim(parent_DEG_list)

parent_up_list<-parent_DEG_list%>%
  filter(log2FoldChange>0)

dim(parent_up_list)

parent_down_list<-parent_DEG_list%>%
  filter(log2FoldChange<0)

dim(parent_down_list)
```
601 total genes with 236 up regulated and 365 down regulated. We are keeping all contrasts in these lists and we will run a loop to conduct enrichment for each contrast.    

## Conduct enrichment for up regulated genes 

Make a list of contrasts to loop through.  

```{r}
levels(as.factor(parent_up_list$contrast))
parent_contrasts<-c("BvsWT", "NBvsWT", "NBvsB")
```
Conduct functional enrichment for each contrast using goseq and rrvgo. Compare against all genes detected in the dataset. I am choosing to compare against genes in our dataset because we are looking at one developmental stage (larvae). Therefore, I do not want to conduct enrichment of all genes in the genome because many functions including calcification and gametogenesis do not apply to the biological function of larvae.  

Conduct at biological process level. Filter for adjusted p-value <0.05 because we have a lot of DEGs.  
```{r}
for (parent in parent_contrasts) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(filtered_Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(filtered_Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- parent_up_list%>%
      filter(contrast==parent)%>%
      pull(gene)
    
    ##Get a list of GO Terms for each constrast
    GO.terms <- filtered_Mcap.annot%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.05
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("output/rna_seq/functional_enrichment/goseq_parent_up_", parent, ".csv"))
    
}
```
Only B vs NB has significant GO term enrichment.  

Create an empty dataframe to populate (only do this once).  

```{r}
parent_go_results <- data.frame()
```

Run a loop to filter enrichment for each temperature contrast using rrvgo. 
```{r}
parent_contrasts<-c("NBvsB")

for (parent in parent_contrasts) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("output/rna_seq/functional_enrichment/goseq_parent_up_", parent, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log10(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$direction<-c("upregulated")
    go_results$contrast<-parent
    
    #add to data frame 
    parent_go_df <- go_results
    parent_go_results <- rbind(parent_go_results, parent_go_df) 

}
```

```{r}
length(unique(parent_go_results$GOterm))
```
3 total enriched GO terms.  

There won't be NB vs WT or B vs WT comparison here, there are no enriched GO terms for this comparison that are upregulated.  

## Conduct enrichment for down regulated genes 

Make a list of contrasts to loop through.  

```{r}
parent_contrasts<-c("BvsWT", "NBvsWT", "NBvsB")
```

Conduct at biological process level. Filter for adjusted p-value <0.05. 
```{r}
for (parent in parent_contrasts) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(filtered_Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(filtered_Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- parent_down_list%>%
      filter(contrast==parent)%>%
      pull(gene)
    
    ##Get a list of GO Terms for each constrast
    GO.terms <- filtered_Mcap.annot%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.05
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("output/rna_seq/functional_enrichment/goseq_parent_down_", parent, ".csv"))
    
}
```

Run a loop to filter enrichment for each temperature contrast using rrvgo.   
```{r}
parent_contrasts<-c("BvsWT", "NBvsWT", "NBvsB")

for (parent in parent_contrasts) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("output/rna_seq/functional_enrichment/goseq_parent_down_", parent, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log10(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$direction<-c("downregulated")
    go_results$contrast<-parent
    
    #add to data frame 
    parent_go_df <- go_results
    parent_go_results <- rbind(parent_go_results, parent_go_df) 

}
```

```{r}
length(unique(parent_go_results$GOterm))
```
There are now 3 enriched GO terms (3 were upregulated, same 3 down regulated). 

## Plot enrichment for up and down regulated genes for each contrast 

Upregulated genes 
```{r}
parent_go_plot_up <- parent_go_results %>% 
  filter(direction=="upregulated")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=contrast, y=term, colour=over_represented_pvalue, size=percHits, fill=over_represented_pvalue)) +
  facet_grid(ParentTerm ~ ., scales="free", labeller = label_wrap_gen(multi_line = TRUE))+
  geom_point()+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  scale_x_discrete(limits=c("BvsWT", "NBvsWT", "NBvsB"))+
  theme_classic() +
  ylab("GO Term")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Enrichment: Upregulated Parent DEGs'); parent_go_plot_up

ggsave(plot=parent_go_plot_up, "figures/rna_seq/functional_enrichment/parent_GO_upregulated.png", height=8, width=8, limitsize=FALSE)
```

Downregulated genes 
```{r}
parent_go_plot_down <- parent_go_results %>% 
  filter(direction=="downregulated")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=contrast, y=term, colour=over_represented_pvalue, size=percHits)) +
  facet_grid(ParentTerm ~ ., scales="free", labeller = label_wrap_gen(multi_line = TRUE))+
  geom_point()+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  theme_classic() +
  scale_x_discrete(limits=c("BvsWT", "NBvsWT", "NBvsB"))+
  ylab("GO Term")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Enrichment: Downregulated Parent DEGs'); parent_go_plot_down

ggsave(plot=parent_go_plot_down, "figures/rna_seq/functional_enrichment/parent_GO_downregulated.png", height=6, width=8, limitsize=FALSE)
```

Plot with up and down regulated with facet by comparison. 

```{r}
parent_go_plot_parentA<-parent_go_results%>%
  mutate(percHits=(numDEInCat/numInCat)*100)%>%
  
  ggplot(aes(x=contrast, y=term, colour=direction)) +
  facet_grid(~ direction, scales="free_y", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=percHits))+
  scale_size(name="Num. GO Terms", breaks=c(2,10,20,50,80), range = c(1, 5))+
  scale_colour_manual(values=c("blue", "red"))+
  scale_x_discrete(limits=c("BvsWT", "NBvsWT", "NBvsB"))+
  theme_classic() +
  xlab("")+
  ylab("GO Term")+
  ggtitle('Parent DEGs')+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"),
        axis.text.x = element_text(size=12, angle=45, vjust=0.5))+
      panel_border(color="black"); parent_go_plot_parentA


ggsave(plot=parent_go_plot_parentA, "figures/rna_seq/functional_enrichment/parent_GO_parent.png", height=6, width=8, limitsize=FALSE)
```

# Interaction DEGs 

DEGs between parent at elevated temperature. 

## Add in annotation information

Merge in relevant annotation information to the DEG list file.  

```{r}
interaction_DEG_list<-left_join(interaction_DEG_list, filtered_Mcap.annot, by="gene")
```

## Plot fold change of temperature DEGs

```{r}
interaction_deg_FC <- interaction_DEG_list %>%
  mutate(label=paste0(Description, "-", gene))%>%
  dplyr::arrange(log2FoldChange) %>%
  
  ggplot(aes(x=log2FoldChange, y=reorder(label, -log2FoldChange), colour=padj)) +
  facet_grid(~contrast, scales="free_y")+
  geom_point(size=2)+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  geom_segment(aes(x=0, xend=log2FoldChange, y=label, yend=label),linewidth=1) +
  geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("Gene - Description")+
  theme(legend.position = "right",
        #axis.text.y = element_text(size=8),
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Fold Change: Parent DEGs at 30-33°C'); interaction_deg_FC

ggsave(plot=interaction_deg_FC, "figures/rna_seq/functional_enrichment/interaction_DEG_foldchange.png", height=8, width=14)
```

## Prepare up and down regulated dataframes 

```{r}
dim(interaction_DEG_list)

interaction_up_list<-interaction_DEG_list%>%
  filter(log2FoldChange>0)

dim(interaction_up_list)

interaction_down_list<-interaction_DEG_list%>%
  filter(log2FoldChange<0)

dim(interaction_down_list)
```
56 total genes with 30 up regulated and 26 down regulated. We are keeping all contrasts in these lists and we will run a loop to conduct enrichment for each contrast.    

## Conduct enrichment for up regulated genes 

Make a list of contrasts to loop through.  

```{r}
levels(as.factor(interaction_up_list$contrast))
interaction_contrasts<-c("NBvsB_30C", "NBvsB_33C")
```
Conduct functional enrichment for each contrast using goseq and rrvgo. Compare against all genes detected in the dataset. I am choosing to compare against genes in our dataset because we are looking at one developmental stage (larvae). Therefore, I do not want to conduct enrichment of all genes in the genome because many functions including calcification and gametogenesis do not apply to the biological function of larvae.  

Conduct at biological process level. Filter for adjusted p-value <0.01 because we have a lot of DEGs.  
```{r}
for (interaction in interaction_contrasts) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(filtered_Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(filtered_Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- interaction_up_list%>%
      filter(contrast==interaction)%>%
      pull(gene)
    
    ##Get a list of GO Terms 
    GO.terms <- filtered_Mcap.annot%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("output/rna_seq/functional_enrichment/goseq_interaction_up_", interaction, ".csv"))
    
}
```
No upregulated GO terms. 

Create an empty dataframe to populate (only do this once).  

```{r}
interaction_go_results <- data.frame()
```

Run a loop to filter enrichment for each temperature contrast using rrvgo. 
```{r}
interaction_contrasts<-c("NBvsB_30C", "NBvsB_33C")

for (interaction in interaction_contrasts) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("output/rna_seq/functional_enrichment/goseq_interaction_up_", interaction, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      #filter(numInCat>1)%>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log10(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$direction<-c("upregulated")
    go_results$contrast<-interaction
    
    #add to data frame 
    interaction_go_df <- go_results
    interaction_go_results <- rbind(interaction_go_results, interaction_go_df) 

}
```

```{r}
length(unique(interaction_go_results$GOterm))
```
0 total enriched GO terms.  

## Conduct enrichment for down regulated genes 

Make a list of contrasts to loop through.  

```{r}
interaction_contrasts<-c("NBvsB_30C", "NBvsB_33C")
```

Conduct at biological process level. Filter for adjusted p-value <0.05. 
```{r}
for (interaction in interaction_contrasts) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(filtered_Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(filtered_Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- interaction_down_list%>%
      filter(contrast==interaction)%>%
      pull(gene)
    
    ##Get a list of GO Terms for each constrast
    GO.terms <- filtered_Mcap.annot%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("output/rna_seq/functional_enrichment/goseq_interaction_down_", interaction, ".csv"))
    
}
```
Only enriched go terms for NB vs B at 33°C. 

Run a loop to filter enrichment for each temperature contrast using rrvgo.   
```{r}
interaction_contrasts<-c("NBvsB_33C")

for (interaction in interaction_contrasts) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("output/rna_seq/functional_enrichment/goseq_interaction_down_", interaction, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log10(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results #%>%
      #filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$direction<-c("downregulated")
    go_results$contrast<-interaction
    
    #add to data frame 
    interaction_go_df <- go_results
    interaction_go_results <- rbind(interaction_go_results, interaction_go_df) 

}
```

```{r}
length(unique(interaction_go_results$GOterm))
```
There are now 3 enriched GO terms (0 were upregulated, leaving 3 downregulated). 

## Plot enrichment for up and down regulated genes for each contrast 

Upregulated genes 
```{r}
interaction_go_plot_up <- interaction_go_results %>% 
  filter(direction=="upregulated")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(hitsPerc=numDEInCat*100/numInCat) %>% 
  
  ggplot(aes(x=contrast, y=GOterm, colour=over_represented_pvalue)) +
  #facet_grid(ParentTerm ~ ., scales="free", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=hitsPerc))+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (5)", range = c(0.1, 2))+
  scale_x_discrete(limits=c("BvsWT", "NBvsWT", "NBvsB"))+
  theme_classic() +
  ylab("GO Term")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Enrichment: Upregulated Parent DEGs at 30-33°C'); interaction_go_plot_up

#ggsave(plot=interaction_go_plot_up, "figures/rna_seq/functional_enrichment/interaction_GO_upregulated.png", height=40, width=6, limitsize=FALSE)
```

Downregulated genes 
```{r}
interaction_go_plot_down <- interaction_go_results %>% 
  filter(direction=="downregulated")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  mutate(hitsPerc=numDEInCat*100/numInCat) %>% 
  
  ggplot(aes(x=contrast, y=term, colour=over_represented_pvalue)) +
  #facet_grid(ParentTerm ~ ., scales="free", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=hitsPerc))+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  scale_size(name="Hits (%)",range = c(1,5))+
  theme_classic() +
  ylab("GO Term")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Enrichment: Downregulated Parent DEGs at 30-33°C'); interaction_go_plot_down

ggsave(plot=interaction_go_plot_down, "figures/rna_seq/functional_enrichment/interaction_GO_downregulated.png", height=6, width=8, limitsize=FALSE)
```

Plot with up and down regulated with facet by comparison. 

```{r}
interaction_go_plot_parentA<-interaction_go_results%>%
  mutate(hitsPerc=numDEInCat*100/numInCat) %>% 
  
  ggplot(aes(x=contrast, y=term, colour=direction)) +
  facet_grid(~ direction, scales="free_y", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=hitsPerc))+
  scale_size(name="Hits (%)", range = c(1, 5))+
  scale_colour_manual(values=c("blue", "red"))+
  theme_classic() +
  xlab("")+
  ylab("Parent Term")+
  ggtitle('NBvsB at 30-33°C DEGs')+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"),
        axis.text.x = element_text(size=12, angle=45, vjust=0.5))+
      panel_border(color="black"); interaction_go_plot_parentA

ggsave(plot=interaction_go_plot_parentA, "figures/rna_seq/functional_enrichment/interaction_GO_parent.png", height=6, width=6, limitsize=FALSE)
```


