---
title: "RNAseq Differential Gene Expression Analysis"
author: "Ariana S Huffmyer"
date: "4/21/2024"
output: html_document
editor_options: 
  chunk_output_type: console
---

Analysis of RNAseq Differential Gene Expression (DEG).  

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# The following setting is important, do not omit.
options(stringsAsFactors = FALSE) #Set Strings to character
```

# Set up 

Load required libraries.  

```{r}
#if ("tidyverse" %in% rownames(installed.packages()) == 'FALSE') install.packages('tidyverse') 
#if ("genefilter" %in% rownames(installed.packages()) == 'FALSE') install.packages('genefilter') 
#if ("DESeq2" %in% rownames(installed.packages()) == 'FALSE') install.packages('DESeq2') 
#if ("RColorBrewer" %in% rownames(installed.packages()) == 'FALSE') install.packages('RColorBrewer') 
#if ("WGCNA" %in% rownames(installed.packages()) == 'FALSE') install.packages('WGCNA') 
#if ("flashClust" %in% rownames(installed.packages()) == 'FALSE') install.packages('flashClust') 
#if ("gridExtra" %in% rownames(installed.packages()) == 'FALSE') install.packages('gridExtra') 
#if ("ComplexHeatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('ComplexHeatmap') 
#if ("goseq" %in% rownames(installed.packages()) == 'FALSE') install.packages('goseq') 
#if ("dplyr" %in% rownames(installed.packages()) == 'FALSE') install.packages('dplyr') 
#if ("clusterProfiler" %in% rownames(installed.packages()) == 'FALSE') install.packages('clusterProfiler') 
#if ("pheatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('pheatmap') 
#if ("magrittr" %in% rownames(installed.packages()) == 'FALSE') install.packages('magrittr') 
#if ("vegan" %in% rownames(installed.packages()) == 'FALSE') install.packages('vegan') 

#BiocManager::install('apeglm')
#BiocManager::install('EnhancedVolcano')
#BiocManager::install("VennDetail")

library("tidyverse")
library("genefilter")
library("WGCNA")
library("DESeq2")
library("RColorBrewer")
library("flashClust")
library("gridExtra")
library("goseq")
library("dplyr")
library("clusterProfiler")
library("pheatmap")
library("magrittr")
library("vegan")
#library("apeglm")
library("EnhancedVolcano")
library("cowplot")
library("viridis")
library("pheatmap")
library("VennDetail")
```

# Data input and filtering  

Import the data files.  

```{r}
#load metadata sheet with sample name and developmental stage information
metadata <- read.csv("data/rna_seq/sample_rnaseq_metadata.csv", header = TRUE, sep = ",")%>%dplyr::select(sample, temperature, parent, symbiont, phenotype)
metadata$code<-paste0(metadata$temperature, "_", metadata$parent)
head(metadata)

#load gene count matrix generated from cluster computation
gcount <- as.data.frame(read.csv("data/rna_seq/Mcapitata2023_gene_count_matrix_STAR.csv", row.names="gene_id"), colClasses = double)
head(gcount)
```

Check that there are no genes with 0 counts across all samples. 

```{r}
dim(gcount) 

gcount<-gcount %>%
     mutate(Total = rowSums(.[, 1:54]))%>%
    filter(!Total==0)%>%
    dplyr::select(!Total)

dim(gcount)
```

We started with 54,384 genes. which is the total number in the annotation. About 6,000 genes that had 0's across all samples (not detected) with 48332 remaining.

Conduct data filtering, this includes:  

*pOverA*: Specifying the minimum count for a proportion of samples for each gene. Here, we are using a pOverA of 0.05. This is because we have 54 samples with a minimum of n=6 samples  per group. Therefore, we will accept genes that are present in 6/54 = 0.11 of the samples because we expect different expression by treatment group (parent x temperature). We are further setting the minimum count of genes to 10, such that 11% of the samples must have a gene count of >10 in order for the gene to remain in the data set.  

Filter in the package "genefilter". Pre-filtering our dataset to reduce the memory size dataframe, increase the speed of the transformation and testing functions, and improve quality of statistical analysis by removing low-coverage counts. Removed counts could represent outliers in the data and removing these improves sensitivity of statistical tests.

```{r}
filt <- filterfun(pOverA(0.1,10))

#create filter for the counts data
gfilt <- genefilter(gcount, filt)

#identify genes to keep by count filter
gkeep <- gcount[gfilt,]

#identify gene lists
gn.keep <- rownames(gkeep)

#gene count data filtered in PoverA, P percent of the samples have counts over A
gcount_filt <- as.data.frame(gcount[which(rownames(gcount) %in% gn.keep),])

#How many rows do we have before and after filtering?
nrow(gcount) #Before
nrow(gcount_filt) #After
```

Before filtering, we had approximately 48,332 genes. After filtering for pOverA, we have approximately 28,613 genes. This indicates that there were 28,613 genes present in 10% of samples at <10 counts per gene. 6/54 = 0.11, so I selected 10% filter because I want to keep genes that are uniquely expressed in one treatment group (n=6 out of 54).  

Prepare sample names. Keep R and any numbers following R in the colnames of gcount_filt.  

```{r}
# Function to extract 'R' followed by numbers from column names
extract_R_numbers <- function(col_names) {
  sub("R(\\d+).*", "\\1", col_names)
}

# Apply the function to all column names
new_col_names <- sapply(names(gcount_filt), extract_R_numbers)

# Rename the columns
names(gcount_filt) <- paste0("R", new_col_names)

# remove "trim." from the name
names(gcount_filt) <- gsub("trim\\.", "", names(gcount_filt))

# Output the modified dataframe
names(gcount_filt)
length(names(gcount_filt))
```

Column names are now correct and there are 54 as expected.  

Print the sample names in the metadata file. 

```{r}
metadata$sample
```

In order for the DESeq2 algorithms to work, the SampleIDs on the metadata file and count matrices have to match exactly and in the same order. They are.  

# Format metadata and count data columns  

Convert temperature and metadata categories to factors. 
```{r}
metadata$temperature<-as.factor(metadata$temperature)
metadata$parent<-as.factor(metadata$parent)
metadata$symbiont<-as.factor(metadata$symbiont)
metadata$code<-as.factor(metadata$code)
```

Set levels of factors. 
```{r}
metadata$temperature<-factor(metadata$temperature, levels=c("27", "30", "33"))
metadata$parent<-factor(metadata$parent, levels=c("Wildtype", "Bleached", "Nonbleached"))
metadata$code<-factor(metadata$code, levels=c("27_Wildtype", "27_Bleached", "27_Nonbleached", "30_Wildtype", "30_Bleached", "30_Nonbleached", "33_Wildtype", "33_Bleached", "33_Nonbleached"))
```

Make sure the metadata and the columns in the gene count matrix are all the same.  
```{r}
metadata$sample
colnames(gcount_filt)
```

Reorder them automatically if needed.  
```{r}
list<-colnames(gcount_filt)
list<-as.factor(list)

metadata$sample<-as.factor(metadata$sample)

# Re-order the levels
metadata$sample <- factor(as.character(metadata$sample), levels=list)
# Re-order the data.frame
metadata_ordered <- metadata[order(metadata$sample),]
metadata_ordered$sample

head(metadata_ordered)

metadata_ordered$sample
colnames(gcount_filt)
head(gcount_filt)
```

These match we are good to continue to the next step!    

# Construct DESeq2 data set by temperature x parent  

Create a DESeqDataSet design from gene count matrix and labels. Here we set the design to look at temperature to test for any differences in gene expression across temperatures. 

The model we will use will analyze the effect of temperature, parent, and their interaction on gene expression. We do not have random or batch effects in this study to analyze.  
```{r}
#Set DESeq2 design
gdds <- DESeqDataSetFromMatrix(countData = gcount_filt,
                              colData = metadata_ordered,
                             design = ~temperature + parent + temperature:parent)
```

First we are going to log-transform the data using a variance stabilizing transforamtion (VST). This is only for visualization purposes. Essentially, this is roughly similar to putting the data on the log2 scale. It will deal with the sampling variability of low counts by calculating within-group variability (if blind=FALSE). Importantly, it does not use the design to remove variation in the data, and so can be used to examine if there may be any variability do to technical factors such as extraction batch effects.

This function calculates a variance stabilizing transformation (VST) from the fitted dispersion-mean relation(s) and then transforms the count data (normalized by division by the size factors or normalization factors), yielding a matrix of values which are now approximately homoskedastic (having constant variance along the range of mean values).    

To do this we first need to calculate the size factors of our samples. This is a rough estimate of how many reads each sample contains compared to the others. In order to use VST (the faster log2 transforming process) to log-transform our data, the size factors need to be less than 4.  

To do this we first need to calculate the size factors of our samples. This is a rough estimate of how many reads each sample contains compared to the others. In order to use VST (the faster log2 transforming process) to log-transform our data, the size factors need to be less than 4.

Chunk should return TRUE if <4.  
```{r}
SF.gdds <- estimateSizeFactors(gdds) #estimate size factors to determine if we can use vst  to transform our data. Size factors should be less than 4 for us to use vst
print(sizeFactors(SF.gdds)) #View size factors

all(sizeFactors(SF.gdds)) < 4
```

All size factors are less than 4, so we can use VST transformation.  

```{r}
gvst <- vst(gdds, blind=FALSE) #apply a variance stabilizing transforamtion to minimize effects of small counts and normalize wrt library siz
head(assay(gvst), 3) #view transformed gene count data for the first three genes in the dataset. 
dim(gvst)
```

# Plot a heatmap of sample to sample distances    

```{r}
gsampleDists <- dist(t(assay(gvst))) #calculate distance matix
gsampleDistMatrix <- as.matrix(gsampleDists) #distance matrix
rownames(gsampleDistMatrix) <- colnames(gvst) #assign row names
colnames(gsampleDistMatrix) <- NULL #assign col names
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255) #assign colors

save_pheatmap_pdf <- function(x, filename, width=7, height=7) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}

pht<-pheatmap(gsampleDistMatrix, #plot matrix
         clustering_distance_rows=gsampleDists, #cluster rows
         clustering_distance_cols=gsampleDists, #cluster columns
         col=colors) #set colors

save_pheatmap_pdf(pht, "figures/rna_seq/sample_distances.pdf")
dev.off()
```

R93 and R57 are outliers, remove 

```{r}
metadata_ordered<-metadata_ordered%>%
  filter(!sample=="R93")%>%
  filter(!sample=="R57")

gcount_filt<-gcount_filt%>%
  select(!R93)%>%
  select(!R57)
```


```{r}
#Set DESeq2 design
gdds <- DESeqDataSetFromMatrix(countData = gcount_filt,
                              colData = metadata_ordered,
                             design = ~temperature + parent + temperature:parent)
```

Chunk should return TRUE if <4.  
```{r}
SF.gdds <- estimateSizeFactors(gdds) #estimate size factors to determine if we can use vst  to transform our data. Size factors should be less than 4 for us to use vst
print(sizeFactors(SF.gdds)) #View size factors

all(sizeFactors(SF.gdds)) < 4
```

All size factors are less than 4, so we can use VST transformation.  

```{r}
gvst <- vst(gdds, blind=FALSE) #apply a variance stabilizing transforamtion to minimize effects of small counts and normalize wrt library size
head(assay(gvst), 3) #view transformed gene count data for the first three genes in the dataset. 
dim(gvst)
```

52 total samples 

Plot a heatmap of sample to sample distances with the outlier removed    
```{r}
gsampleDists <- dist(t(assay(gvst))) #calculate distance matix
gsampleDistMatrix <- as.matrix(gsampleDists) #distance matrix
rownames(gsampleDistMatrix) <- colnames(gvst) #assign row names
colnames(gsampleDistMatrix) <- NULL #assign col names
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255) #assign colors

save_pheatmap_pdf <- function(x, filename, width=7, height=7) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}

pht<-pheatmap(gsampleDistMatrix, #plot matrix
         clustering_distance_rows=gsampleDists, #cluster rows
         clustering_distance_cols=gsampleDists, #cluster columns
         col=colors) #set colors

save_pheatmap_pdf(pht, "figures/rna_seq/sample_distances_outlier.pdf")
dev.off()
```

# Conduct PERMANOVA and PCA for all genes

Export data for PERMANOVA test.  
```{r}
test<-t(assay(gvst)) #export as matrix
test<-as.data.frame(test)

#add category columns
test$sample<-rownames(test)
test$temperature<-metadata$temperature[match(test$sample, metadata$sample)]
test$parent<-metadata$parent[match(test$sample, metadata$sample)]
test$code<-metadata$code[match(test$sample, metadata$sample)]
```

Build PERMANOVA model for all genes in the dataset.  
```{r}
library(vegan)
library(factoextra)

dim(test)
scaled_test <-prcomp(test[c(1:28613)], scale=TRUE, center=TRUE)
fviz_eig(scaled_test)

# scale data
vegan <- scale(test[c(1:28613)])

# PerMANOVA 
permanova<-adonis2(vegan ~ parent*temperature, data = test, method='eu')
permanova
```

adonis2(formula = vegan ~ parent * temperature, data = test, method = "eu")
                   Df SumOfSqs      R2      F Pr(>F)    
parent              2   207617 0.14228 5.3258  0.001 ***
temperature         2   323578 0.22174 8.3004  0.001 ***
parent:temperature  4    89923 0.06162 1.1534  0.157    
Residual           43   838145 0.57436                  
Total              51  1459263 1.00000     

Gene expression is significantly different between temperature and parent, no interactive effect detected.  

## Examine PCA and sample distances of all genes  

Plot a PCA of samples by temperature and parental phenotype for all genes.        

```{r}
parent_names<-c("Bleached" = "Susceptible (C)", "Nonbleached" = "Resistant (C/D)", "Wildtype" = "Wildtype (C/D)")

gPCAdata <- plotPCA(gvst, intgroup = c("temperature", "parent"), returnData=TRUE)
percentVar <- round(100*attr(gPCAdata, "percentVar")) #plot PCA of samples with all data

allgenesfilt_PCA <- ggplot(gPCAdata, aes(PC1, PC2, color=parent, shape=temperature)) + 
  geom_point(size=3) + 
  ggtitle(expression(bold("All genes (28,613 genes)")))+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  scale_color_manual(name="Phenotype (Symbiont)", values=c("Wildtype"="gray", "Bleached"="orange", "Nonbleached"="brown4"), labels=parent_names)+
  scale_shape_manual(name="Temperature", values=c(19,17,15))+
  geom_text(x=0, y=8, label="p[Phenotype]=0.001", color="black")+
  geom_text(x=0, y=7, label="p[Temperature]=0.001", color="black")+
  geom_text(x=0, y=6, label="p[Interaction]=0.157", color="darkgray")+
  theme_classic() + #Set background color
  ggtitle("A")+
  theme(panel.border = element_blank(), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()); allgenesfilt_PCA

ggsave("figures/rna_seq/pca.pdf", allgenesfilt_PCA, width=8, height=4)
ggsave("figures/rna_seq/pca.jpeg", allgenesfilt_PCA, width=8, height=4)
```

Add a plot with elipses.  

```{r}
allgenes_ellipse<-allgenesfilt_PCA + stat_ellipse();allgenes_ellipse

ggsave("figures/rna_seq/pca_ellipse.pdf", allgenes_ellipse, width=9, height=5)
ggsave("figures/rna_seq/pca_ellipse.jpeg", allgenes_ellipse, width=9, height=5)
```

# Run DEG analysis of temperature, parent, and interaction   

Use the likelihood ratio approach based on guidance from DESeq2: "DESeq2 offers two kinds of hypothesis tests: the Wald test, where we use the estimated standard error of a log2 fold change to test if it is equal to zero, and the likelihood ratio test (LRT). The LRT examines two models for the counts, a full model with a certain number of terms and a reduced model, in which some of the terms of the full model are removed. The test determines if the increased likelihood of the data using the extra terms in the full model is more than expected if those extra terms are truly zero.

The LRT is therefore useful for testing multiple terms at once, for example testing 3 or more levels of a factor at once, or all interactions between two variables. The LRT for count data is conceptually similar to an analysis of variance (ANOVA) calculation in linear regression, except that in the case of the Negative Binomial GLM, we use an analysis of deviance (ANODEV), where the deviance captures the difference in likelihood between a full and a reduced model." 

## 1. View DEG's by Temperature 

Run Wald test.  
```{r, message = FALSE}
DEG_T <- DESeq(gdds, test="Wald")
```

View results by temperature effects. Create dataframe for each comparison and add a column for the specific contrast.  
```{r}
resultsNames(DEG_T)
#this shows the comparisons made (shown similar to lm output in terms of interactions)
#The interaction terms temperature33.parentBleached and temperature33.parentNonbleached give the difference between the parental effect for a given temperature and the parental effect for the reference parent (Wildtype)

temp30_deg<-results(DEG_T, contrast=c("temperature","30","27"))
temp30_deg <- as.data.frame(subset(temp30_deg, padj<0.05))
#temp30_deg <- as.data.frame(subset(temp30_deg, abs(log2FoldChange)>1.5))
temp30_deg$contrast <- c("30vs27")
temp30_deg$gene <- rownames(temp30_deg)
rownames(temp30_deg) <- NULL

dim(temp30_deg)
#2162 DEGS

temp33_deg<-results(DEG_T, contrast=c("temperature","33","27"))
temp33_deg <- as.data.frame(subset(temp33_deg, padj<0.05))
#temp33_deg <- as.data.frame(subset(temp33_deg, abs(log2FoldChange)>1.5))
temp33_deg$contrast <- c("33vs27")
temp33_deg$gene <- rownames(temp33_deg)
rownames(temp33_deg) <- NULL

dim(temp33_deg)
#9098 DEGs

temp30vs33_deg<-results(DEG_T, contrast=c("temperature","33","30"))
temp30vs33_deg <- as.data.frame(subset(temp30vs33_deg, padj<0.05))
#temp30vs33_deg <- as.data.frame(subset(temp30vs33_deg, abs(log2FoldChange)>1.5))
temp30vs33_deg$contrast <- c("33vs30")
temp30vs33_deg$gene <- rownames(temp30vs33_deg)
rownames(temp30vs33_deg) <- NULL

dim(temp30vs33_deg)
#5867 DEGs
```

In these contrasts, positive fold changes indicate higher expression in the first listed contrast.  

Combine this list of DEGs into one dataframe. 

```{r}
DEG_temperature<-rbind(temp30_deg, temp33_deg, temp30vs33_deg)
dim(DEG_temperature)
length(unique(DEG_temperature$gene))
```
There are 17,127 total genes and 9715 unique genes, indicating that some are shared in the contrasts. 

Get a VST normalized gene count matrix for these DEGs. 
```{r}
DEG_temperature_vst <- gdds[unique(DEG_temperature$gene)]
dim(DEG_temperature_vst)

DEG_temperature_vst <- varianceStabilizingTransformation(DEG_temperature_vst) 
```

### Upset plot of genes between tempeature contrasts 

Create an upset diagram showing genes unique and shared in each temperature comparison.  

```{r}
list1<-DEG_temperature%>%
  filter(contrast=="30vs27")

list2<-DEG_temperature%>%
  filter(contrast=="33vs27")

list3<-DEG_temperature%>%
  filter(contrast=="33vs30")

ven <- venndetail(list("30vs27" = list1$gene, "33vs27" = list2$gene,
                    "33vs30" = list3$gene))

upset_temperature<-plot(ven, type = "upset");upset_temperature


pdf("figures/rna_seq/upset_temperature.pdf", width=8, height=6)
par(mar = c(5, 5, 5, 5))
plot(ven, type = "upset")
dev.off()
```

Most are different between 33C and the two lower temperatures. 

### Plot a heatmap for genes that are differentially expressed between temperatures.

Set themes and metadata.  
```{r}
legend_names_col = colnames(assay(DEG_temperature_vst))

df <- as.data.frame(colData(gdds)[,c("temperature","parent")])
df$temperature <- factor(as.character(df$temperature), levels=c("27", "30", "33"))
df$parent <- factor(as.character(df$parent), levels=c("Wildtype", "Bleached", "Nonbleached"))
df$phenotype<-if_else(df$parent=="Bleached", "Susceptible (C)", 
                  if_else(df$parent=="Nonbleached", "Resistant (C/D)", 
                          if_else(df$parent=="Wildtype", "Wildtype (C/D)", NA)))
df<-df%>%
  arrange(temperature, phenotype)%>%
  select(!parent)

temp_colors <- c("Ambient" = "#2c7bb6",   # calm blue
                 "Moderate (+3)" = "#7b3294", # stressed purple
                 "High (+6)" = "#d53e4f") 

ann_colors = list(
    temperature = c("27" = "#2c7bb6", "30" = "#7b3294","33"= "#d53e4f"), 
    phenotype = c("Susceptible (C)" = "orange", "Resistant (C/D)" = "brown4", "Wildtype (C/D)" = "gray")
)
```

Plot heatmap.  
```{r}
pdf("figures/rna_seq/DEG_heatmap_temperature.pdf", width=15, height=20)
pheatmap(assay(DEG_temperature_vst), cluster_rows=TRUE, show_rownames=FALSE, color=inferno(10), show_colnames=TRUE, fontsize_row=3, scale="row", cluster_cols=TRUE, annotation_col=df, annotation_colors = ann_colors, labels_col=legend_names_col)
dev.off()
```

### Run a PERMANOVA and plot a PCA for temperature comparisons.  

Conduct PERMANOVA by parent and temperature 

Export data for PERMANOVA test.  
```{r}
test_DEG_temp<-t(assay(DEG_temperature_vst)) #export as matrix
test_DEG_temp<-as.data.frame(test_DEG_temp)

#add category columns
test_DEG_temp$sample<-rownames(test_DEG_temp)
test_DEG_temp$temperature<-metadata$temperature[match(test_DEG_temp$sample, metadata$sample)]
test_DEG_temp$parent<-metadata$parent[match(test_DEG_temp$sample, metadata$sample)]
```

Build PERMANOVA model for DEG's that are different between temperatures.  
```{r}
dim(test_DEG_temp)

scaled_test_DEG_temp <-prcomp(test_DEG_temp[c(1:9715)], scale=TRUE, center=TRUE)
fviz_eig(scaled_test_DEG_temp)

# scale data
vegan_DEG_temp <- scale(test_DEG_temp[c(1:9715)])

# PerMANOVA 
permanova_DEG<-adonis2(vegan_DEG_temp ~ parent*temperature, data = test_DEG_temp, method='eu')
permanova_DEG
```
About 50% of variance is explained by first PC followed by <10% on following PCs. 

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegan_DEG_temp ~ parent * temperature, data = test_DEG_temp, method = "eu")
                   Df SumOfSqs      R2       F Pr(>F)    
parent              2    60594 0.12230  7.3965  0.001 ***
temperature         2   235575 0.47546 28.7559  0.001 ***
parent:temperature  4    23163 0.04675  1.4137  0.124    
Residual           43   176133 0.35549                   
Total              51   495465 1.00000               

Temperature is significant as expected because we are analyzing temperature DEGs. Parent is also significant. This makes sense because in our global gene expression PERMANOVA and PCA above we see that global expression is different between parental phenotypes. This shows that parental separation is present in differentially expressed genes between temperatures. We will look into this further below.  

Plot a PCA of DEG's for temperature.        
```{r}
gPCAdata_DEG_temp <- plotPCA(DEG_temperature_vst, intgroup = c("temperature", "parent"), returnData=TRUE)
percentVar_DEG_temp <- round(100*attr(gPCAdata_DEG_temp, "percentVar")) #plot PCA of samples with all data

DEG_PCA_temp <- ggplot(gPCAdata_DEG_temp, aes(PC1, PC2, color=parent, shape=temperature)) + 
  geom_point(size=3) + 
  ggtitle(expression(bold("DEGs: temperature (6,849 genes)")))+
  xlab(paste0("PC1: ",percentVar_DEG_temp[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_DEG_temp[2],"% variance")) +
  scale_color_manual(name="Parental Phenotype", values=c("Wildtype"="gray", "Bleached"="orange", "Nonbleached"="brown4"), labels=parent_names)+
  scale_shape_manual(name="Temperature", values=c(19,17,15))+
  geom_text(x=0, y=-5, label="p[Phenotype]=0.002", color="black")+
  geom_text(x=0, y=-6.5, label="p[Temperature]=0.001", color="black")+
  geom_text(x=0, y=-8, label="p[Interaction]=0.201", color="darkgray")+
  theme_classic() + #Set background color
  theme(panel.border = element_blank(), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())+
  stat_ellipse()+
  ggtitle("B"); DEG_PCA_temp

ggsave("figures/rna_seq/pca_DEG_temperature.pdf", DEG_PCA_temp, width=8, height=4)
ggsave("figures/rna_seq/pca_DEG_temperature.png", DEG_PCA_temp, width=8, height=4)
```

## 2. View DEG's by Parent 

```{r, message = FALSE}
DEG_P <- DESeq(gdds, test="Wald")
```

View results by parental effects. Create dataframe for each comparison and add a column for the specific contrast.  
```{r}
resultsNames(DEG_P)
#this shows the comparisons made (shown similar to lm output in terms of interactions)
#The interaction terms temperature33.parentBleached and temperature33.parentNonbleached give the difference between the parental effect for a given temperature and the parental effect for the reference parent (Wildtype)

BvsWT_deg<-results(DEG_P, contrast=c("parent","Bleached","Wildtype"))
BvsWT_deg <- as.data.frame(subset(BvsWT_deg, padj<0.05))
#BvsWT_deg <- as.data.frame(subset(BvsWT_deg, abs(log2FoldChange)>1.5))
BvsWT_deg$contrast <- c("BvsWT")
BvsWT_deg$gene <- rownames(BvsWT_deg)
rownames(BvsWT_deg) <- NULL

dim(BvsWT_deg)
#2071 DEGs

NBvsWT_deg<-results(DEG_P, contrast=c("parent","Nonbleached","Wildtype"))
NBvsWT_deg <- as.data.frame(subset(NBvsWT_deg, padj<0.05))
#NBvsWT_deg <- as.data.frame(subset(NBvsWT_deg, abs(log2FoldChange)>1.5))
NBvsWT_deg$contrast <- c("NBvsWT")
NBvsWT_deg$gene <- rownames(NBvsWT_deg)
rownames(NBvsWT_deg) <- NULL

dim(NBvsWT_deg)
#2621 DEGs

NBvsB_deg<-results(DEG_P, contrast=c("parent","Nonbleached","Bleached"))
NBvsB_deg <- as.data.frame(subset(NBvsB_deg, padj<0.05))
#NBvsB_deg <- as.data.frame(subset(NBvsB_deg, abs(log2FoldChange)>1.5))
NBvsB_deg$contrast <- c("NBvsB")
NBvsB_deg$gene <- rownames(NBvsB_deg)
rownames(NBvsB_deg) <- NULL

dim(NBvsB_deg)
#2540 DEGs
```
In these contrasts, positive fold changes indicate higher expression in the first listed contrast.  

Combine this list of DEGs into one dataframe. 

```{r}
DEG_parent<-rbind(BvsWT_deg, NBvsWT_deg, NBvsB_deg)
dim(DEG_parent)
length(unique(DEG_parent$gene))
```
There are 7232 total DEGs and 4863 unique genes, indicating that some are shared in the contrasts. 

Get a VST normalized gene count matrix for these DEGs. 
```{r}
DEG_parent_vst <- gdds[unique(DEG_parent$gene)]
dim(DEG_parent_vst)

DEG_parent_vst <- varianceStabilizingTransformation(DEG_parent_vst) 
```

### Upset plot of genes between tempeature contrasts 

Create an upset diagram showing genes unique and shared in each parent comparison.  

```{r}
list1_parent<-DEG_parent%>%
  filter(contrast=="NBvsWT")

list2_parent<-DEG_parent%>%
  filter(contrast=="BvsWT")

list3_parent<-DEG_parent%>%
  filter(contrast=="NBvsB")

ven_parent <- venndetail(list("NBvsWT" = list1_parent$gene, "BvsWT" = list2_parent$gene,
                    "NBvsB" = list3_parent$gene))

upset_parent<-plot(ven_parent, type = "upset");upset_parent


pdf("figures/rna_seq/upset_parent.pdf", width=8, height=6)
par(mar = c(5, 5, 5, 5))
plot(ven_parent, type = "upset")
dev.off()
```
 

### Plot a heatmap for genes that are differentially expressed between parent phenotypes 

Set themes and metadata.  
```{r}
legend_names_col = colnames(assay(DEG_parent_vst))

df <- as.data.frame(colData(gdds)[,c("temperature","parent")])
df$temperature <- factor(as.character(df$temperature), levels=c("27", "30", "33"))
df$parent <- factor(as.character(df$parent), levels=c("Wildtype", "Bleached", "Nonbleached"))
df$phenotype<-if_else(df$parent=="Bleached", "Susceptible (C)", 
                  if_else(df$parent=="Nonbleached", "Resistant (C/D)", 
                          if_else(df$parent=="Wildtype", "Wildtype (C/D)", NA)))
df<-df%>%
  arrange(temperature, phenotype)%>%
  select(!parent)

temp_colors <- c("Ambient" = "#2c7bb6",   # calm blue
                 "Moderate (+3)" = "#7b3294", # stressed purple
                 "High (+6)" = "#d53e4f") 

ann_colors = list(
    temperature = c("27" = "#2c7bb6", "30" = "#7b3294","33"= "#d53e4f"), 
    phenotype = c("Susceptible (C)" = "orange", "Resistant (C/D)" = "brown4", "Wildtype (C/D)" = "gray")
)
```

Plot heatmap.  
```{r}
pdf("figures/rna_seq/DEG_heatmap_parent.pdf", width=15, height=20)
pheatmap(assay(DEG_parent_vst), cluster_rows=TRUE, show_rownames=FALSE, color=inferno(10), show_colnames=TRUE, fontsize_row=3, scale="row", cluster_cols=TRUE, annotation_col=df, annotation_colors = ann_colors, labels_col=legend_names_col)
dev.off()
```

### Run a PERMANOVA and plot a PCA for parent comparisons.  

Conduct PERMANOVA by parent and temperature of parental DEGs. 

Export data for PERMANOVA test.  
```{r}
test_DEG_parent<-t(assay(DEG_parent_vst)) #export as matrix
test_DEG_parent<-as.data.frame(test_DEG_parent)

#add category columns
test_DEG_parent$sample<-rownames(test_DEG_parent)
test_DEG_parent$temperature<-metadata$temperature[match(test_DEG_parent$sample, metadata$sample)]
test_DEG_parent$parent<-metadata$parent[match(test_DEG_parent$sample, metadata$sample)]
```

Build PERMANOVA model for DEG's that are different between temperatures.  
```{r}
dim(test_DEG_parent)

scaled_test_DEG_parent <-prcomp(test_DEG_parent[c(1:4863)], scale=TRUE, center=TRUE)
fviz_eig(scaled_test_DEG_parent)

# scale data
vegan_DEG_parent <- scale(test_DEG_parent[c(1:4863)])

# PerMANOVA 
permanova_DEG_parent<-adonis2(vegan_DEG_parent ~ parent*temperature, data = test_DEG_parent, method='eu')
permanova_DEG_parent
```
About 25% of variance is explained by first PC followed by 18% on the second PC and 15% on the third PC. 

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegan_DEG_parent ~ parent * temperature, data = test_DEG_parent, method = "eu")
                   Df SumOfSqs      R2       F Pr(>F)    
parent              2    85809 0.34599 21.2935  0.001 ***
temperature         2    62465 0.25186 15.5008  0.001 ***
parent:temperature  4    13098 0.05281  1.6251  0.026 *  
Residual           43    86641 0.34934                   
Total              51   248013 1.00000            

Parent is significant as expected because we are analyzing parental DEGs. There are still temperature effects, which also makes sense because there was such a strong temperature effect on global expression. Interestingly, there is a parent x temperature interaction, indicating response to temperature is mediated by parental phenotype. We will look into this more next.   

Plot a PCA of DEG's for parent        
```{r}
gPCAdata_DEG_parent <- plotPCA(DEG_parent_vst, intgroup = c("temperature", "parent"), returnData=TRUE)
percentVar_DEG_parent <- round(100*attr(gPCAdata_DEG_parent, "percentVar")) #plot PCA of samples with all data

DEG_PCA_parent <- ggplot(gPCAdata_DEG_parent, aes(PC1, PC2, color=parent, shape=temperature)) + 
  geom_point(size=3) + 
  xlab(paste0("PC1: ",percentVar_DEG_parent[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_DEG_parent[2],"% variance")) +
  scale_color_manual(name="Parental Phenotype", values=c("Wildtype"="gray", "Bleached"="orange", "Nonbleached"="brown4"), labels=parent_names)+
  scale_shape_manual(name="Temperature", values=c(19,17,15))+
  geom_text(x=0, y=10, label="p[Phenotype]=0.001", color="black")+
  geom_text(x=0, y=8, label="p[Temperature]=0.001", color="black")+
  geom_text(x=0, y=6, label="p[Interaction]=0.026", color="black")+
  theme_classic() + #Set background color
  theme(panel.border = element_blank(), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())+
  stat_ellipse()+
  ggtitle("D"); DEG_PCA_parent

ggsave("figures/rna_seq/pca_DEG_parent.pdf", DEG_PCA_parent, width=8, height=4)
ggsave("figures/rna_seq/pca_DEG_parent.png", DEG_PCA_parent, width=8, height=4)
```

From this we can see very clear separation by parent driven by separation of the B group. 

## 3. View DEG's differential between parents at elevated temperatures.  

Run differential expression test using LRT testing for significant genes by parent by removing the effect of temperature:parent and comparing the significance of genes in the full model (`parent + temperature + parent:temperature`) to the reduced model removing the interactive effect.   
```{r, message = FALSE}
DEG_I <- DESeq(gdds, test="Wald")
```

Next, view genes that are uniquely differentially expressed between parental types at 30°C and 33°C. In other words, view genes that are different between NB and B groups at 30°C and 33°. Conduct this for 30°C and then 33°C and then join the lists.   

parent_Nonbleached_vs_Wildtype: nonbleached effects
parent_Bleached_vs_Wildtype: bleached effects
temperature30.parentNonbleached: nonbleached effects at 33 
temperature30.parentBleached: bleached effects at 33

The first c(...) inside list() is the A group (Nonbleached at 33 °C)

The second c(...) inside list() is the B group (Bleached at 33 °C)

DESeq2 then calculates (A) – (B) for you.

The formula is then NB - B + temp:NB - temp:B to find the change between B and NB at the temperature. 

The sum = exactly the total log2 fold change between NB and B at 33 °C.

Create dataframe for each comparison and add a column for the specific contrast.  
```{r}
resultsNames(DEG_I)
#this shows the comparisons made (shown similar to lm output in terms of interactions)

NBvsB_30_deg <- results(DEG_I,
   contrast = list(
     c("parent_Nonbleached_vs_Wildtype",
       "temperature30.parentNonbleached"),
     c("parent_Bleached_vs_Wildtype",
       "temperature30.parentBleached")
   ))

#The coefficient "parent_Nonbleached_vs_Bleached" gives you the baseline parent effect (usually at the reference temperature, 27 °C).

#The coefficients "temperature33.parentNonbleached" and "temperature33.parentBleached" give you how much the temp effect at 33 °C deviates from that baseline for each parent type.

NBvsB_30_deg <- as.data.frame(subset(NBvsB_30_deg, padj<0.05))
#NBvsB_30_deg <- as.data.frame(subset(NBvsB_30_deg, abs(log2FoldChange)>1))
NBvsB_30_deg$contrast <- c("NBvsB_30C")
NBvsB_30_deg$gene <- rownames(NBvsB_30_deg)
rownames(NBvsB_30_deg) <- NULL

dim(NBvsB_30_deg)
#2844 DEGs

#NBvsB_33_deg<-results(DEG_I, contrast=list("temperature33.parentNonbleached", "temperature33.parentBleached"))

NBvsB_33_deg <- results(DEG_I,
   contrast = list(
     c("parent_Nonbleached_vs_Wildtype",
       "temperature33.parentNonbleached"),
     c("parent_Bleached_vs_Wildtype",
       "temperature33.parentBleached")
   ))

#The coefficient "parent_Nonbleached_vs_Bleached" gives you the baseline parent effect (usually at the reference temperature, 27 °C).

#The coefficients "temperature33.parentNonbleached" and "temperature33.parentBleached" give you how much the temp effect at 33 °C deviates from that baseline for each parent type.

NBvsB_33_deg <- as.data.frame(subset(NBvsB_33_deg, padj<0.05))
#NBvsB_33_deg <- as.data.frame(subset(NBvsB_33_deg, abs(log2FoldChange)>1))
NBvsB_33_deg$contrast <- c("NBvsB_33C")
NBvsB_33_deg$gene <- rownames(NBvsB_33_deg)
rownames(NBvsB_33_deg) <- NULL

dim(NBvsB_33_deg)
#3853 DEGs
```

Combine this list of DEGs into one dataframe. 

```{r}
DEG_interaction<-rbind(NBvsB_30_deg, NBvsB_33_deg)
dim(DEG_interaction)
length(unique(DEG_interaction$gene))
```
There are 6697 total genes and 5007 unique DEGs.

```{r}
#DEG_interaction_unique <- anti_join(DEG_interaction, DEG_parent, by = "gene")

DEG_interaction_unique<-DEG_interaction

dim(DEG_interaction_unique)
str(DEG_interaction_unique)
length(unique(DEG_interaction_unique$gene))
```
There are 5007 genes that are uniquely differentially expressed between NB and B at elevated temperature (30 or 33°C). 

Get a VST normalized gene count matrix for these DEGs. 
```{r}
DEG_interaction_vst <- gdds[unique(DEG_interaction_unique$gene)]
dim(DEG_interaction_vst)

DEG_interaction_vst <- varianceStabilizingTransformation(DEG_interaction_vst) 
```

### Upset plot of genes between temperature contrasts 

Create an upset diagram showing genes unique and shared in each temperature comparison.  

```{r}
list1_interaction<-DEG_interaction_unique%>%
  filter(contrast=="NBvsB_30C")
dim(list1_interaction)

list2_interaction<-DEG_interaction_unique%>%
  filter(contrast=="NBvsB_33C")
dim(list2_interaction)

ven_interaction <- venndetail(list("NBvsB_30C" = list1_interaction$gene, "NBvsB_33C" = list2_interaction$gene))

upset_interaction<-plot(ven_interaction, type = "upset");upset_interaction

pdf("figures/rna_seq/upset_interaction.pdf", width=8, height=6)
plot(upset_interaction, type = "upset")
dev.off()
```

### Plot a heatmap for genes that are differentially expressed between parent phenotypes at elevated temperature 

Set themes and metadata.  
```{r}
legend_names_col = colnames(assay(DEG_interaction_vst))

df <- as.data.frame(colData(gdds)[,c("temperature","parent")])
df$temperature <- factor(as.character(df$temperature), levels=c("27", "30", "33"))
df$parent <- factor(as.character(df$parent), levels=c("Wildtype", "Bleached", "Nonbleached"))
df$phenotype<-if_else(df$parent=="Bleached", "Susceptible (C)", 
                  if_else(df$parent=="Nonbleached", "Resistant (C/D)", 
                          if_else(df$parent=="Wildtype", "Wildtype (C/D)", NA)))
df<-df%>%
  arrange(temperature, phenotype)%>%
  select(!parent)

temp_colors <- c("Ambient" = "#2c7bb6",   # calm blue
                 "Moderate (+3)" = "#7b3294", # stressed purple
                 "High (+6)" = "#d53e4f") 

ann_colors = list(
    temperature = c("27" = "#2c7bb6", "30" = "#7b3294","33"= "#d53e4f"), 
    phenotype = c("Susceptible (C)" = "orange", "Resistant (C/D)" = "brown4", "Wildtype (C/D)" = "gray")
)
```

Plot heatmap.  
```{r}
pdf("figures/rna_seq/DEG_heatmap_interaction.pdf", width=8, height=8)
pheatmap(assay(DEG_interaction_vst), cluster_rows=TRUE, show_rownames=FALSE, color=inferno(10), show_colnames=TRUE, fontsize_row=3, scale="row", cluster_cols=TRUE, annotation_col=df, annotation_colors = ann_colors, labels_col=legend_names_col)
dev.off()
```

### Run a PERMANOVA and plot a PCA for interaction comparisons

Conduct PERMANOVA by parent and temperature of interaction DEGs. 

Export data for PERMANOVA test.  
```{r}
test_DEG_interaction<-t(assay(DEG_interaction_vst)) #export as matrix
test_DEG_interaction<-as.data.frame(test_DEG_interaction)

#add category columns
test_DEG_interaction$sample<-rownames(test_DEG_interaction)
test_DEG_interaction$temperature<-metadata$temperature[match(test_DEG_interaction$sample, metadata$sample)]
test_DEG_interaction$parent<-metadata$parent[match(test_DEG_interaction$sample, metadata$sample)]
```

Build PERMANOVA model for DEG's that are different between NB and B at 30-33°C 
```{r}
dim(test_DEG_interaction)

scaled_test_DEG_interaction <-prcomp(test_DEG_interaction[c(1:5007)], scale=TRUE, center=TRUE)
fviz_eig(scaled_test_DEG_interaction)

# scale data
vegan_DEG_interaction <- scale(test_DEG_interaction[c(1:5007)])

# PerMANOVA 
permanova_DEG_interaction<-adonis2(vegan_DEG_interaction ~ parent*temperature, data = test_DEG_interaction, method='eu')
permanova_DEG_interaction
```
About 25% of variance is explained by first PC followed by 20% on the second PC and 10% on the third PC. 

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegan_DEG_interaction ~ parent * temperature, data = test_DEG_interaction, method = "eu")
                   Df SumOfSqs      R2       F Pr(>F)    
parent              2    81975 0.32102 18.0796  0.001 ***
temperature         2    59791 0.23414 13.1869  0.001 ***
parent:temperature  4    16109 0.06308  1.7764  0.009 ** 
Residual           43    97483 0.38175                   
Total              51   255357 1.00000                   

Interaction is significant as expected. Strong parent and temperature effects are also present as expected.   

Plot a PCA of DEG's for the interaction

```{r}
gPCAdata_DEG_interaction <- plotPCA(DEG_interaction_vst, intgroup = c("temperature", "parent"), returnData=TRUE)
percentVar_DEG_interaction <- round(100*attr(gPCAdata_DEG_interaction, "percentVar")) #plot PCA of samples with all data

DEG_PCA_interaction <- ggplot(gPCAdata_DEG_interaction, aes(PC1, PC2, color=parent, shape=temperature)) + 
  geom_point(size=3) + 
  ggtitle("F")+
  xlab(paste0("PC1: ",percentVar_DEG_parent[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_DEG_parent[2],"% variance")) +
  scale_color_manual(name="Parental Phenotype", values=c("Wildtype"="gray", "Bleached"="orange", "Nonbleached"="brown4"))+
  scale_shape_manual(name="Temperature", values=c(19,17,15))+
  geom_text(x=-2, y=-2.75, label="p[Parent]=0.001", color="black")+
  geom_text(x=-2, y=-4.25, label="p[Temperature]=0.001", color="black")+
  geom_text(x=-2, y=-5.75, label="p[Interaction]=0.010", color="black")+
  theme_classic() + #Set background color
  theme(panel.border = element_blank(), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank())+
  stat_ellipse(); DEG_PCA_interaction

ggsave("figures/rna_seq/pca_DEG_interaction.pdf", DEG_PCA_interaction, width=8, height=4)
ggsave("figures/rna_seq/pca_DEG_interaction.png", DEG_PCA_interaction, width=8, height=4)
```

# Save lists of gene counts and DEG information 

## Temperature

DEG_temperature: List of DEG's and fold change/pvalues 
DEG_temperature_vst: Count matrix of genes from DEG list 

```{r}
deg_temperature_list<-DEG_temperature%>%
  dplyr::select(gene, everything())
write_csv(deg_temperature_list, "output/rna_seq/temperature_deg_list.csv")

dim(DEG_temperature_vst)
deg_temperature_counts<-t(assay(DEG_temperature_vst))
deg_temperature_counts<-as.data.frame(deg_temperature_counts)
dim(deg_temperature_counts)
deg_temperature_counts$sample<-rownames(deg_temperature_counts)
deg_temperature_counts<-deg_temperature_counts%>%
  dplyr::select(sample, everything())
write_csv(deg_temperature_counts, "output/rna_seq/temperature_deg_counts.csv")
```

## Parent

DEG_parent: List of DEG's and fold change/pvalues 
DEG_parent_vst: Count matrix of genes from DEG list 

```{r}
deg_parent_list<-DEG_parent%>%
  dplyr::select(gene, everything())
write_csv(deg_parent_list, "output/rna_seq/parent_deg_list.csv")

dim(DEG_parent_vst)
deg_parent_counts<-t(assay(DEG_parent_vst))
deg_parent_counts<-as.data.frame(deg_parent_counts)
dim(deg_parent_counts)
deg_parent_counts$sample<-rownames(deg_parent_counts)
deg_parent_counts<-deg_parent_counts%>%
  dplyr::select(sample, everything())
write_csv(deg_parent_counts, "output/rna_seq/parent_deg_counts.csv")
```

## NB vs B at elevated temperature

DEG_interaction_unique: List of DEG's and fold change/pvalues
DEG_interaction_vst: Count matrix of genes from DEG list 

```{r}
dim(DEG_interaction_unique)
deg_interaction_list<-DEG_interaction_unique%>%
  dplyr::select(gene, everything())
write_csv(deg_interaction_list, "output/rna_seq/NBvsB_30-33_deg_list.csv")

dim(DEG_interaction_vst)
deg_interaction_counts<-t(assay(DEG_interaction_vst))
deg_interaction_counts<-as.data.frame(deg_interaction_counts)
dim(deg_interaction_counts)
deg_interaction_counts$sample<-rownames(deg_interaction_counts)
deg_interaction_counts<-deg_interaction_counts%>%
  dplyr::select(sample, everything())
write_csv(deg_interaction_counts, "output/rna_seq/NBvsB_30-33_deg_counts.csv")
```

## All genes detected 

List of all genes in the dataset (gvst) - genes which will serve as a reference for functional enrichment.   

```{r}
dim(gvst)
all_genes<-t(assay(gvst))
all_genes<-as.data.frame(all_genes)
dim(all_genes)
all_genes$sample<-rownames(all_genes)
all_genes<-all_genes%>%
  dplyr::select(sample, everything())

write_csv(file="output/rna_seq/all_genes.csv", all_genes)
```

