---
title: "RNAseq DEG Functional Enrichment Analysis"
author: "Ariana S Huffmyer"
date: "4/21/2024"
output: html_document
editor_options: 
  chunk_output_type: console
---

LEFT OFF WITH ADDING DF FOR UP AND DOWN REGULATED GENES 




Analysis of RNAseq Differential Gene Expression (DEG).  

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# The following setting is important, do not omit.
options(stringsAsFactors = FALSE) #Set Strings to character
```

# Set up 

```{r}
if ("tidyverse" %in% rownames(installed.packages()) == 'FALSE') install.packages('tidyverse') 
if ("genefilter" %in% rownames(installed.packages()) == 'FALSE') install.packages('genefilter') 
if ("DESeq2" %in% rownames(installed.packages()) == 'FALSE') install.packages('DESeq2') 
if ("RColorBrewer" %in% rownames(installed.packages()) == 'FALSE') install.packages('RColorBrewer') 
if ("WGCNA" %in% rownames(installed.packages()) == 'FALSE') install.packages('WGCNA') 
if ("flashClust" %in% rownames(installed.packages()) == 'FALSE') install.packages('flashClust') 
if ("gridExtra" %in% rownames(installed.packages()) == 'FALSE') install.packages('gridExtra') 
if ("ComplexHeatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('ComplexHeatmap') 
if ("goseq" %in% rownames(installed.packages()) == 'FALSE') install.packages('goseq') 
if ("dplyr" %in% rownames(installed.packages()) == 'FALSE') install.packages('dplyr') 
if ("clusterProfiler" %in% rownames(installed.packages()) == 'FALSE') install.packages('clusterProfiler') 
if ("pheatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('pheatmap') 
if ("magrittr" %in% rownames(installed.packages()) == 'FALSE') install.packages('magrittr') 
if ("rtracklayer" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rtracklayer")
if ("GenomicRanges" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GenomicRanges")
if ("plyranges" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("plyranges")
if ("GSEABase" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GSEABase")
if ("GOSim" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GOSim")
if ("stats" %in% rownames(installed.packages()) == 'FALSE') install.packages("stats")
if ("ggdendro" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("ggdendro")
if ("GO.db" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GO.db")
if ("rrvgo" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rrvgo")

#BiocManager::install("org.Ce.eg.db", force=TRUE) #install if needed 

library("tidyverse")
library("genefilter")
library("DESeq2")
library("RColorBrewer")
library("WGCNA")
library("flashClust")
library("gridExtra")
library("ComplexHeatmap")
library("goseq")
library("dplyr")
library("clusterProfiler")
library("pheatmap")
library("magrittr")
library("rtracklayer")
library("GenomicRanges")
library("plyranges")
library("GSEABase")
library("GOSim")
library("stats")
library("ggdendro")
library("GO.db")
library("rrvgo")
```

# Read in DEG and gene files 

1. Read in file with vst transformed counts of all genes detected and kept after filtering. 
```{r}
all_genes<-read.csv("output/rna_seq/all_genes.csv")
```

2. Read in file of DEG's between parental phenotypes
```{r}
parent_DEGs_up<-read.csv("output/rna_seq/parent_DEGs_up.csv")
parent_DEGs_down<-read.csv("output/rna_seq/parent_DEGs_down.csv")
```

3. Read in file of DEG's between temperatures 
```{r}
temp_DEGs<-read.csv("output/rna_seq/temp_DEGs.csv")
```

4. Read in file of DEG's between NB and B parental phenotypes uniquely differential at 33Â°C 
```{r}
unique_NB_33_DEGs<-read.csv("output/rna_seq/unique_NBvsB_33C_DEGs.csv")
```

# Prepare functional annotation files  

Downloaded functional annotation (version 3) from: http://cyanophora.rutgers.edu/montipora/Montipora_capitata_HIv3.genes.EggNog_results.txt.gz (EggNog) and http://cyanophora.rutgers.edu/montipora/Montipora_capitata_HIv3.genes.KEGG_results.txt.gz (KEGG). 

Scripts to generate this file are from Mcapitata 2020 project scripts here: https://github.com/AHuffmyer/EarlyLifeHistory_Energetics 

Downloaded on 21 April 2024. Unzipped on desktop. I then removed the # in #query in the first column, otherwise it does not read in column names.    

```{r}
Mcap.annot <- read.table("data/rna_seq/Montipora_capitata_HIv3.genes.EggNog_results.txt",  quote="", sep="\t", header=TRUE)
dim(Mcap.annot)

# column names
head(Mcap.annot)
tail(Mcap.annot)
```

This functional annotation as 24,072 genes. This is 44% of total protein predicted genes described in the publication associated with this annotation. https://academic.oup.com/gigascience/article/doi/10.1093/gigascience/giac098/6815755 

We had 24,005 genes in our dataset that were detected and present at counts >10. 

## Make a data frame that connects genes, traits, and modules  

Match up genes in datExpr file to annotation file
```{r}
names(Mcap.annot)

#remove sample column name 
probes = names(all_genes)
probes <- gsub("sample", "", probes)
probes <- probes[probes != ""]

probes2annot = match(probes, Mcap.annot$query) #match genes in datExpr to genes in annotation file, note I removed the # before query in this file before loading

# The following is the number of probes without annotation 
sum(is.na(probes2annot))

row_nas<-which(is.na(probes2annot))

#view the genes that do not have a match in the annotation file
missing<-as.data.frame(probes[row_nas])
print(missing)
```

9389 genes without annotation information. Spot check manually in the annotation file confirms the annotation file does not contain these genes.       

Create the starting data frame
```{r}
names(Mcap.annot)

geneInfo = data.frame(gene_id = probes, #add gene id
Accession = Mcap.annot$seed_ortholog[probes2annot], #add accession number
Bitscore = Mcap.annot$score[probes2annot], #add bitscore
eValue = Mcap.annot$evalue[probes2annot], #add e value
Description = Mcap.annot$Description[probes2annot], #add description of gene
Annotation.GO.ID = Mcap.annot$GOs[probes2annot]) #add GO ID's

head(geneInfo)
```

Add KEGG annotation information.  Downloaded from: http://cyanophora.rutgers.edu/montipora/Montipora_capitata_HIv3.genes.KEGG_results.txt.gz on 21 April 2024. 

```{r}
kegg<-read.table("data/rna_seq/Montipora_capitata_HIv3.genes.KEGG_results.txt", sep="", quote="", na.strings=c("","NA"), blank.lines.skip = FALSE, header=FALSE)

head(kegg)
```

Add KEGG annotations to each gene.  
```{r}
geneInfo$KEGG<-kegg$V2[match(geneInfo$gene_id, kegg$V1)]
```

Save geneInfo as a CSV
```{r}
head(geneInfo)
dim(geneInfo)

geneInfo$Annotation.GO.ID <- gsub(";NA", "", geneInfo$Annotation.GO.ID) #Remove NAs
geneInfo$Annotation.GO.ID <- gsub("NA", "", geneInfo$Annotation.GO.ID) #Remove NAs
dim(geneInfo)

write.csv(geneInfo, file = "output/rna_seq/geneInfo.csv") #gene info for reference/supplement

```

# Format DEG lists  

Format our gene vst data frames for plotting.  

1. all_genes
```{r}
all_genes_expr<-as.matrix(all_genes)
all_genes_expr<-t(all_genes_expr)
all_genes_expr<-as.data.frame(all_genes_expr, row.names = rownames(all_genes_expr), make.names=TRUE)

colnames(all_genes_expr)<-all_genes_expr[1,]

all_genes_expr<-all_genes_expr[-1,]

dim(all_genes_expr)
str(all_genes_expr)

# Convert the matrix to a data frame
char_df <- as.data.frame(all_genes_expr)

# Convert the data frame to numeric
numeric_df <- as.data.frame(lapply(char_df, as.numeric))

# Convert the numeric data frame back to a matrix
all_genes_expr_num <- as.matrix(numeric_df)

# Restore row and column names
rownames(all_genes_expr_num) <- rownames(all_genes_expr)
colnames(all_genes_expr_num) <- colnames(all_genes_expr)

dim(all_genes_expr_num)
names(all_genes_expr_num)
head(all_genes_expr_num)
```
We now have 24005 genes in rows with 54 samples in columns. 

2. parent_DEGs

Upregulated 
```{r}
parent_DEGs_expr_up<-as.matrix(parent_DEGs_up)
parent_DEGs_expr_up<-t(parent_DEGs_expr_up)
parent_DEGs_expr_up<-as.data.frame(parent_DEGs_expr_up, row.names = rownames(parent_DEGs_expr_up), make.names=TRUE)

colnames(parent_DEGs_expr_up)<-parent_DEGs_expr_up[1,]

parent_DEGs_expr_up<-parent_DEGs_expr_up[-1,]

dim(parent_DEGs_expr_up)

# Convert the matrix to a data frame
char_df <- as.data.frame(parent_DEGs_expr_up)

# Convert the data frame to numeric
numeric_df <- as.data.frame(lapply(char_df, as.numeric))

# Convert the numeric data frame back to a matrix
parent_DEGs_expr_num_up <- as.matrix(numeric_df)

# Restore row and column names
rownames(parent_DEGs_expr_num_up) <- rownames(parent_DEGs_expr_up)
colnames(parent_DEGs_expr_num_up) <- colnames(parent_DEGs_expr_up)

dim(parent_DEGs_expr_num_up)
```
We now have 618 upregulated genes in rows with 54 samples in columns. 

Downregulated 
```{r}
parent_DEGs_expr_down<-as.matrix(parent_DEGs_down)
parent_DEGs_expr_down<-t(parent_DEGs_expr_down)
parent_DEGs_expr_down<-as.data.frame(parent_DEGs_expr_down, row.names = rownames(parent_DEGs_expr_down), make.names=TRUE)

colnames(parent_DEGs_expr_down)<-parent_DEGs_expr_down[1,]

parent_DEGs_expr_down<-parent_DEGs_expr_down[-1,]

dim(parent_DEGs_expr_down)

# Convert the matrix to a data frame
char_df <- as.data.frame(parent_DEGs_expr_down)

# Convert the data frame to numeric
numeric_df <- as.data.frame(lapply(char_df, as.numeric))

# Convert the numeric data frame back to a matrix
parent_DEGs_expr_num_down <- as.matrix(numeric_df)

# Restore row and column names
rownames(parent_DEGs_expr_num_down) <- rownames(parent_DEGs_expr_down)
colnames(parent_DEGs_expr_num_down) <- colnames(parent_DEGs_expr_down)

dim(parent_DEGs_expr_num_down)
```
There are 494 down regulated genes with 54 samples.  


3. temp_DEGs
```{r}
temp_DEGs_expr<-as.matrix(temp_DEGs)
temp_DEGs_expr<-t(temp_DEGs_expr)
temp_DEGs_expr<-as.data.frame(temp_DEGs_expr, row.names = rownames(temp_DEGs_expr), make.names=TRUE)

colnames(temp_DEGs_expr)<-temp_DEGs_expr[1,]

temp_DEGs_expr<-temp_DEGs_expr[-1,]

dim(temp_DEGs_expr)

# Convert the matrix to a data frame
char_df <- as.data.frame(temp_DEGs_expr)

# Convert the data frame to numeric
numeric_df <- as.data.frame(lapply(char_df, as.numeric))

# Convert the numeric data frame back to a matrix
temp_DEGs_expr_num <- as.matrix(numeric_df)

# Restore row and column names
rownames(temp_DEGs_expr_num) <- rownames(temp_DEGs_expr)
colnames(temp_DEGs_expr_num) <- colnames(temp_DEGs_expr)

dim(temp_DEGs_expr_num)
names(temp_DEGs_expr_num)
head(temp_DEGs_expr_num)
```
We now have 2533 genes in rows with 54 samples in columns. 

3. unique_NB_33_DEGs
```{r}
unique_NB_33_DEGs_expr<-as.matrix(unique_NB_33_DEGs)
unique_NB_33_DEGs_expr<-t(unique_NB_33_DEGs_expr)
unique_NB_33_DEGs_expr<-as.data.frame(unique_NB_33_DEGs_expr, row.names = rownames(unique_NB_33_DEGs_expr), make.names=TRUE)

colnames(unique_NB_33_DEGs_expr)<-unique_NB_33_DEGs_expr[1,]

unique_NB_33_DEGs_expr<-unique_NB_33_DEGs_expr[-1,]

dim(unique_NB_33_DEGs_expr)

# Convert the matrix to a data frame
char_df <- as.data.frame(unique_NB_33_DEGs_expr)

# Convert the data frame to numeric
numeric_df <- as.data.frame(lapply(char_df, as.numeric))

# Convert the numeric data frame back to a matrix
unique_NB_33_DEGs_expr_num <- as.matrix(numeric_df)

# Restore row and column names
rownames(unique_NB_33_DEGs_expr_num) <- rownames(unique_NB_33_DEGs_expr)
colnames(unique_NB_33_DEGs_expr_num) <- colnames(unique_NB_33_DEGs_expr)

dim(unique_NB_33_DEGs_expr_num)
names(unique_NB_33_DEGs_expr_num)
head(unique_NB_33_DEGs_expr_num)

str(unique_NB_33_DEGs_expr_num)
```
We now have 474 genes in rows with 54 samples in columns. 

# Get gene length information   

```{r}
#import file
gff <- rtracklayer::import("data/rna_seq/Montipora_capitata_HIv3.genes_fixed.gff3") #if this doesn't work, restart R and try again 

transcripts <- subset(gff, type == "transcript") #keep only transcripts 

transcripts_gr <- makeGRangesFromDataFrame(transcripts, keep.extra.columns=TRUE) #extract length information

transcript_lengths <- width(transcripts_gr) #isolate length of each transcript

seqnames<-transcripts_gr$ID #extract list of gene id 

lengths<-cbind(seqnames, transcript_lengths)

lengths<-as.data.frame(lengths) #convert to data frame
```

Add in length to the gene info file .    

```{r}
geneInfo$Length<-lengths$transcript_lengths[match(geneInfo$gene_id, lengths$seqnames)]
```

Format GO terms to remove dashes and quotes.  
```{r}
#remove dashes, remove quotes from columns and separate by semicolons (replace , with ;) in KEGG_ko, KEGG_pathway, and Annotation.GO.ID columns
geneInfo$Annotation.GO.ID <- gsub(",", ";", geneInfo$Annotation.GO.ID)
geneInfo$Annotation.GO.ID <- gsub('"', "", geneInfo$Annotation.GO.ID)
geneInfo$Annotation.GO.ID <- gsub("-", NA, geneInfo$Annotation.GO.ID)
```


# Functional Enrichment with Cluster Profiler 

## 1. Parent DEGs 

### Parent DEG's - Upregulated 

```{r}
dim(parent_DEGs_expr_num_up)
```
618 upregulated genes in NB compared to B group. 

Generate a list of genes in the parental DEG list. 
```{r}
parent_up_list<-rownames(parent_DEGs_expr_num_up)
```

Generate dataframes for parental DEGs 
```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
    
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
    
### Generate vector with names in just the module we are analyzing
ID.vector.parent.up <- geneInfo%>%
      filter(gene_id %in% parent_up_list)%>%
      pull(gene_id)
    
##Get a list of GO Terms 
GO.terms.parent.up <- geneInfo%>%
    filter(gene_id %in% parent_up_list)%>%
    dplyr::select(gene_id, Annotation.GO.ID)
    
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms.parent.up$Annotation.GO.ID), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms.parent.up$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "Annotation.GO.ID")
GO.terms.parent.up<-split2

##Construct list of genes with 1 for genes in parental list and 0 for genes not in the parental list
gene.vector.parent.up=as.integer(ALL.vector %in% ID.vector.parent.up) 
names(gene.vector.parent.up)<-ALL.vector#set names

#weight gene vector by bias for length of gene 
pwf.parent.up<-nullp(gene.vector.parent.up, ID.vector.parent.up, bias.data=LENGTH.vector) 
    
#run goseq using Wallenius method for all categories of GO terms 
GO.wall.parent.up<-goseq(pwf.parent.up, ID.vector.parent.up, gene2cat=GO.terms.parent.up, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)

GO.parent.up <- GO.wall.parent.up[order(GO.wall.parent.up$over_represented_pvalue),]
colnames(GO.parent.up)[1] <- "GOterm"
    
#adjust p-values 
GO.parent.up$bh_adjust <-  p.adjust(GO.parent.up$over_represented_pvalue, method="BH") #add BH adjusted p-values
  
#Filtering for p < 0.01
GO.parent.up <- GO.parent.up %>%
    dplyr::filter(bh_adjust<0.01) %>%
    dplyr::arrange(., ontology, bh_adjust)

dim(GO.parent.up)

#Write file of results 
write_csv(GO.parent.up, file = paste0("output/rna_seq/goseq_parent_DEGs_up.csv"))
```

Plot enrichment for parental upregulated GO terms.  
```{r}
go_results.parent.up<-GO.parent.up%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)

#Reduce/collapse GO term set with the rrvgo package 
simMatrix.parent.up <- calculateSimMatrix(go_results.parent.up$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
#calculate similarity 
scores.parent.up <- setNames(-log(go_results.parent.up$bh_adjust), go_results.parent.up$GOterm)

reducedTerms.parent.up <- reduceSimMatrix(simMatrix.parent.up,
                                scores.parent.up,
                                threshold=0.9,
                                orgdb="org.Ce.eg.db")
    
#keep only the goterms from the reduced list
go_results.parent.up<-go_results.parent.up%>%
      filter(GOterm %in% reducedTerms.parent.up$go)
    
#add in parent terms to list of go terms 
go_results.parent.up$ParentTerm<-reducedTerms.parent.up$parentTerm[match(go_results.parent.up$GOterm, reducedTerms.parent.up$go)]
      
#plot significantly enriched GO terms by Slim Category faceted by slim term 
GO.plot.parent.up <-go_results.parent.up%>%
  mutate(hitsPerc=numDEInCat*100/numInCat) %>% 
  
  ggplot(aes(x = hitsPerc, y = term, colour=bh_adjust)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank());GO.plot.parent.up

ggsave(filename=paste0("figures/rna_seq/Parent_GOenrich_BP_up.png"), plot=GO.plot.parent.up, dpi=300, width=8, height=20, units="in")

```










### Parent DEG's - Downregulated 

```{r}
parent_DEGs_expr_num_down
```













# Run GO Seq for Parent DEGs

Generate a list of genes in the parental DEG list. 
```{r}
parent_list<-rownames(parent_DEGs_expr_num)
```

Generate dataframes for parental DEGs 
```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
    
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
    
### Generate vector with names in just the module we are analyzing
ID.vector.parent <- geneInfo%>%
      filter(gene_id %in% parent_list)%>%
      pull(gene_id)
    
##Get a list of GO Terms 
GO.terms.parent <- geneInfo%>%
    filter(gene_id %in% parent_list)%>%
    dplyr::select(gene_id, Annotation.GO.ID)
    
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms.parent$Annotation.GO.ID), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms.parent$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "Annotation.GO.ID")
GO.terms.parent<-split2

##Construct list of genes with 1 for genes in parental list and 0 for genes not in the parental list
gene.vector.parent=as.integer(ALL.vector %in% ID.vector.parent) 
names(gene.vector.parent)<-ALL.vector#set names

#weight gene vector by bias for length of gene 
pwf.parent<-nullp(gene.vector.parent, ID.vector.parent, bias.data=LENGTH.vector) 
    
#run goseq using Wallenius method for all categories of GO terms 
GO.wall.parent<-goseq(pwf.parent, ID.vector.parent, gene2cat=GO.terms.parent, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)

GO.parent <- GO.wall.parent[order(GO.wall.parent$over_represented_pvalue),]
colnames(GO.parent)[1] <- "GOterm"
    
#adjust p-values 
GO.parent$bh_adjust <-  p.adjust(GO.parent$over_represented_pvalue, method="BH") #add adjusted p-values
  
#Filtering for p < 0.01
GO.parent <- GO.parent %>%
    dplyr::filter(bh_adjust<0.01) %>%
    dplyr::arrange(., ontology, bh_adjust)

dim(GO.parent)

#Write file of results 
write_csv(GO.parent, file = paste0("output/rna_seq/goseq_parent_DEGs.csv"))
```

Plot enrichment for parental GO terms.  
```{r}
go_results.parent<-GO.parent%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)

#Reduce/collapse GO term set with the rrvgo package 
   simMatrix.parent <- calculateSimMatrix(go_results.parent$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
#calculate similarity 
  scores.parent <- setNames(-log(go_results.parent$bh_adjust), go_results.parent$GOterm)
  reducedTerms.parent <- reduceSimMatrix(simMatrix.parent,
                                scores.parent,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
#keep only the goterms from the reduced list
    go_results.parent<-go_results.parent%>%
      filter(GOterm %in% reducedTerms.parent$go)
    
#add in parent terms to list of go terms 
go_results.parent$ParentTerm<-reducedTerms.parent$parentTerm[match(go_results.parent$GOterm, reducedTerms.parent$go)]

#go_results.parent<-go_results.parent%>%
 #     filter(grepl(paste(keywords, collapse="|"), ParentTerm))
      
#plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot.parent <-  ggplot(go_results.parent, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank());GO.plot.parent

ggsave(filename=paste0("figures/rna_seq/Parent_GOenrich_BP.png"), plot=GO.plot.parent, dpi=300, width=8, height=49, units="in")

```

# Run GO Seq for temp DEGs

Generate a list of genes in the temp DEG list. 
```{r}
temp_list<-rownames(temp_DEGs_expr_num)
```

Generate dataframes for temp DEGs 
```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
    
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
    
### Generate vector with names in just the module we are analyzing
ID.vector.temp <- geneInfo%>%
      filter(gene_id %in% temp_list)%>%
      pull(gene_id)
    
##Get a list of GO Terms 
GO.terms.temp <- geneInfo%>%
    filter(gene_id %in% temp_list)%>%
    dplyr::select(gene_id, Annotation.GO.ID)
    
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms.temp$Annotation.GO.ID), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms.temp$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "Annotation.GO.ID")
GO.terms.temp<-split2

##Construct list of genes with 1 for genes in parental list and 0 for genes not in the parental list
gene.vector.temp=as.integer(ALL.vector %in% ID.vector.temp) 
names(gene.vector.temp)<-ALL.vector#set names

#weight gene vector by bias for length of gene 
pwf.temp<-nullp(gene.vector.temp, ID.vector.temp, bias.data=LENGTH.vector) 
    
#run goseq using Wallenius method for all categories of GO terms 
GO.wall.temp<-goseq(pwf.temp, ID.vector.temp, gene2cat=GO.terms.temp, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)

GO.temp <- GO.wall.temp[order(GO.wall.temp$over_represented_pvalue),]
colnames(GO.temp)[1] <- "GOterm"
    
#adjust p-values 
GO.temp$bh_adjust <-  p.adjust(GO.temp$over_represented_pvalue, method="BH") #add adjusted p-values
  
#Filtering for p < 0.01
GO.temp <- GO.temp %>%
    dplyr::filter(bh_adjust<0.0001) %>%
    dplyr::arrange(., ontology, bh_adjust)

dim(GO.temp)

#Write file of results 
write_csv(GO.temp, file = paste0("output/rna_seq/goseq_temp_DEGs.csv"))
```

Plot enrichment for temp GO terms.  
```{r}
go_results.temp<-GO.temp%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)

#Reduce/collapse GO term set with the rrvgo package 
simMatrix.temp <- calculateSimMatrix(go_results.temp$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
#calculate similarity 
scores.temp <- setNames(-log(go_results.temp$bh_adjust), go_results.temp$GOterm)
reducedTerms.temp <- reduceSimMatrix(simMatrix.temp,
                                scores.temp,
                                threshold=0.9,
                                orgdb="org.Ce.eg.db")
    
#keep only the goterms from the reduced list
go_results.temp<-go_results.temp%>%
      filter(GOterm %in% reducedTerms.temp$go)
    
#add in temp terms to list of go terms 
go_results.temp$ParentTerm<-reducedTerms.temp$parentTerm[match(go_results.temp$GOterm, reducedTerms.temp$go)]

#plot significantly enriched GO terms by Slim Category faceted by slim term 
GO.plot.temp <-  ggplot(go_results.temp, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank());GO.plot.temp

ggsave(filename=paste0("figures/rna_seq/Temp_GOenrich_BP.png"), plot=GO.plot.temp, dpi=300, width=8, height=49, units="in")

```


# Run GO Seq for unique NB vs B at 33Â°C DEGs

Generate a list of genes in the unique DEG list. 
```{r}
unique_list<-rownames(unique_NB_33_DEGs_expr_num)
```

Generate dataframes for temp DEGs 
```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
    
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
    
### Generate vector with names in just the module we are analyzing
ID.vector.unique <- geneInfo%>%
      filter(gene_id %in% unique_list)%>%
      pull(gene_id)
    
##Get a list of GO Terms 
GO.terms.unique <- geneInfo%>%
    filter(gene_id %in% unique_list)%>%
    dplyr::select(gene_id, Annotation.GO.ID)
    
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms.unique$Annotation.GO.ID), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms.unique$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "Annotation.GO.ID")
GO.terms.unique<-split2

##Construct list of genes with 1 for genes in parental list and 0 for genes not in the parental list
gene.vector.unique=as.integer(ALL.vector %in% ID.vector.unique) 
names(gene.vector.unique)<-ALL.vector#set names

#weight gene vector by bias for length of gene 
pwf.unique<-nullp(gene.vector.unique, ID.vector.unique, bias.data=LENGTH.vector) 
    
#run goseq using Wallenius method for all categories of GO terms 
GO.wall.unique<-goseq(pwf.unique, ID.vector.unique, gene2cat=GO.terms.unique, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)

GO.unique <- GO.wall.unique[order(GO.wall.unique$over_represented_pvalue),]
colnames(GO.unique)[1] <- "GOterm"
    
#adjust p-values 
GO.unique$bh_adjust <-  p.adjust(GO.unique$over_represented_pvalue, method="BH") #add adjusted p-values
  
#Filtering for p < 0.01
GO.unique <- GO.unique %>%
    dplyr::filter(bh_adjust<0.01) %>%
    dplyr::arrange(., ontology, bh_adjust)

dim(GO.unique)

#Write file of results 
write_csv(GO.unique, file = paste0("output/rna_seq/goseq_NBvsBat33_DEGs.csv"))
```

Plot enrichment for unique GO terms.  
```{r}
go_results.unique<-GO.unique%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)

#Reduce/collapse GO term set with the rrvgo package 
simMatrix.unique <- calculateSimMatrix(go_results.unique$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
#calculate similarity 
scores.unique <- setNames(-log(go_results.unique$bh_adjust), go_results.unique$GOterm)
reducedTerms.unique <- reduceSimMatrix(simMatrix.unique,
                                scores.unique,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
#keep only the goterms from the reduced list
go_results.unique<-go_results.unique%>%
      filter(GOterm %in% reducedTerms.unique$go)
    
#add in unique terms to list of go terms 
go_results.unique$ParentTerm<-reducedTerms.unique$parentTerm[match(go_results.unique$GOterm, reducedTerms.unique$go)]

#plot significantly enriched GO terms by Slim Category faceted by slim term 
GO.plot.unique <-  ggplot(go_results.unique, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    facet_grid(ParentTerm ~ ., scales = "free", space='free', labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank());GO.plot.unique

ggsave(filename=paste0("figures/rna_seq/NBvsB_33C_GOenrich_BP.png"), plot=GO.plot.unique, dpi=300, width=8, height=49, units="in")

```
