---
title: "RNAseq Differential Gene Expression Analysis"
author: "Ariana S Huffmyer"
date: "2/18/2022"
output: html_document
editor_options: 
  chunk_output_type: console
---

Analysis of RNAseq Differential Gene Expression (DEG).  

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# The following setting is important, do not omit.
options(stringsAsFactors = FALSE) #Set Strings to character
```

# Set up 
Load required libraries.  

```{r}
if ("tidyverse" %in% rownames(installed.packages()) == 'FALSE') install.packages('tidyverse') 
if ("genefilter" %in% rownames(installed.packages()) == 'FALSE') install.packages('genefilter') 
if ("DESeq2" %in% rownames(installed.packages()) == 'FALSE') install.packages('DESeq2') 
if ("RColorBrewer" %in% rownames(installed.packages()) == 'FALSE') install.packages('RColorBrewer') 
if ("WGCNA" %in% rownames(installed.packages()) == 'FALSE') install.packages('WGCNA') 
if ("flashClust" %in% rownames(installed.packages()) == 'FALSE') install.packages('flashClust') 
if ("gridExtra" %in% rownames(installed.packages()) == 'FALSE') install.packages('gridExtra') 
if ("ComplexHeatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('ComplexHeatmap') 
if ("goseq" %in% rownames(installed.packages()) == 'FALSE') install.packages('goseq') 
if ("dplyr" %in% rownames(installed.packages()) == 'FALSE') install.packages('dplyr') 
if ("clusterProfiler" %in% rownames(installed.packages()) == 'FALSE') install.packages('clusterProfiler') 
if ("pheatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('pheatmap') 
if ("magrittr" %in% rownames(installed.packages()) == 'FALSE') install.packages('magrittr') 
if ("vegan" %in% rownames(installed.packages()) == 'FALSE') install.packages('vegan') 

library("tidyverse")
library("genefilter")
library("WGCNA")
library("DESeq2")
library("RColorBrewer")
library("flashClust")
library("gridExtra")
library("goseq")
library("dplyr")
library("clusterProfiler")
library("pheatmap")
library("magrittr")
library("vegan")
```

# Data input and filtering  

Import the data files.  

```{r}
#load metadata sheet with sample name and developmental stage information
metadata <- read.csv("data/rna_seq/sample_rnaseq_metadata.csv", header = TRUE, sep = ",")%>%select(sample, temperature, parent, symbiont, phenotype)
metadata$code<-paste0(metadata$temperature, "_", metadata$parent)
head(metadata)

#load gene count matrix generated from cluster computation
gcount <- as.data.frame(read.csv("data/rna_seq/Mcapitata2023_gene_count_matrix.csv", row.names="gene_id"), colClasses = double)
head(gcount)
```

Check that there are no genes with 0 counts across all samples. 

```{r}
dim(gcount) 

gcount<-gcount %>%
     mutate(Total = rowSums(.[, 1:54]))%>%
    filter(!Total==0)%>%
    select(!Total)

dim(gcount)
```

We started with 54,384 genes. which is the total number in the annotation. About 10,000 genes that had 0's across all samples (not detected) with 44690 remaining.

Conduct data filtering, this includes:  

*pOverA*: Specifying the minimum count for a proportion of samples for each gene. Here, we are using a pOverA of 0.05. This is because we have 54 samples with a minimum of n=6 samples  per group. Therefore, we will accept genes that are present in 6/54 = 0.11 of the samples because we expect different expression by coral colony We are further setting the minimum count of genes to 10, such that 11% of the samples must have a gene count of >10 in order for the gene to remain in the data set.  

Filter in the package "genefilter". Pre-filtering our dataset to reduce the memory size dataframe, increase the speed of the transformation and testing functions, and improve quality of statistical analysis by removing low-coverage counts. Removed counts could represent outliers in the data and removing these improves sensitivity of statistical tests.

```{r}
filt <- filterfun(pOverA(0.1,11))

#create filter for the counts data
gfilt <- genefilter(gcount, filt)

#identify genes to keep by count filter
gkeep <- gcount[gfilt,]

#identify gene lists
gn.keep <- rownames(gkeep)

#gene count data filtered in PoverA, P percent of the samples have counts over A
gcount_filt <- as.data.frame(gcount[which(rownames(gcount) %in% gn.keep),])

#How many rows do we have before and after filtering?
nrow(gcount) #Before
nrow(gcount_filt) #After
```

Before filtering, we had approximately 44,690 genes. After filtering for pOverA, we have approximately 24,005 genes. This indicates that there were 24,005 genes present in 10% of samples at <10 counts per gene.  

Prepare sample names. Keep R and any numbers following R in the colnames of gcount_filt.  

```{r}
# Function to extract 'R' followed by numbers from column names
extract_R_numbers <- function(col_names) {
  sub("R(\\d+).*", "\\1", col_names)
}

# Apply the function to all column names
new_col_names <- sapply(names(gcount_filt), extract_R_numbers)

# Rename the columns
names(gcount_filt) <- paste0("R", new_col_names)

# Output the modified dataframe
names(gcount_filt)
length(names(gcount_filt))
```

Column names are now correct and there are 54 as expected.  

Print the sample names in the metadata file. 

```{r}
metadata$sample
```

In order for the DESeq2 algorithms to work, the SampleIDs on the metadata file and count matrices have to match exactly and in the same order. They are.  

# Format metadata and count data columns  

Convert temperature and metadata categories to factors. 
```{r}
metadata$temperature<-as.factor(metadata$temperature)
metadata$parent<-as.factor(metadata$parent)
metadata$symbiont<-as.factor(metadata$symbiont)
metadata$code<-as.factor(metadata$code)
```

Set levels of factors. 
```{r}
metadata$temperature<-factor(metadata$temperature, levels=c("27", "30", "33"))
metadata$parent<-factor(metadata$parent, levels=c("Wildtype", "Bleached", "Nonbleached"))
metadata$code<-factor(metadata$code, levels=c("27_Wildtype", "27_Bleached", "27_Nonbleached", "30_Wildtype", "30_Bleached", "30_Nonbleached", "33_Wildtype", "33_Bleached", "33_Nonbleached"))
```

Make sure the metadata and the columns in the gene count matrix are all the same.  
```{r}
metadata$sample
colnames(gcount_filt)
```

Reorder them automatically if needed.  
```{r}
list<-colnames(gcount_filt)
list<-as.factor(list)

metadata$sample<-as.factor(metadata$sample)

# Re-order the levels
metadata$sample <- factor(as.character(metadata$sample), levels=list)
# Re-order the data.frame
metadata_ordered <- metadata[order(metadata$sample),]
metadata_ordered$sample

head(metadata_ordered)

metadata_ordered$sample
colnames(gcount_filt)
head(gcount_filt)
```

These match we are good to continue to the next step!    

# Construct DESeq2 data set by temperature x parent  

Create a DESeqDataSet design from gene count matrix and labels. Here we set the design to look at temperature to test for any differences in gene expression across temperatures  
```{r}
#Set DESeq2 design
gdds <- DESeqDataSetFromMatrix(countData = gcount_filt,
                              colData = metadata_ordered,
                              design = ~temperature + parent + temperature:parent)
```

First we are going to log-transform the data using a variance stabilizing transforamtion (VST). This is only for visualization purposes. Essentially, this is roughly similar to putting the data on the log2 scale. It will deal with the sampling variability of low counts by calculating within-group variability (if blind=FALSE). Importantly, it does not use the design to remove variation in the data, and so can be used to examine if there may be any variability do to technical factors such as extraction batch effects.

To do this we first need to calculate the size factors of our samples. This is a rough estimate of how many reads each sample contains compared to the others. In order to use VST (the faster log2 transforming process) to log-transform our data, the size factors need to be less than 4.

Chunk should return TRUE if <4.  
```{r}
SF.gdds <- estimateSizeFactors(gdds) #estimate size factors to determine if we can use vst  to transform our data. Size factors should be less than 4 for us to use vst
print(sizeFactors(SF.gdds)) #View size factors

all(sizeFactors(SF.gdds)) < 4
```

All size factors are less than 4, so we can use VST transformation.  

```{r}
gvst <- vst(gdds, blind=FALSE) #apply a variance stabilizing transforamtion to minimize effects of small counts and normalize wrt library size
head(assay(gvst), 3) #view transformed gene count data for the first three genes in the dataset.  
```

# Conduct PERMANOVA by parent and temperature for all genes

Export data for PERMANOVA test.  
```{r}
test<-t(assay(gvst)) #export as matrix
test<-as.data.frame(test)

#add category columns
test$sample<-rownames(test)
test$temperature<-metadata$temperature[match(test$sample, metadata$sample)]
test$parent<-metadata$parent[match(test$sample, metadata$sample)]
test$code<-metadata$code[match(test$sample, metadata$sample)]

```

Build PERMANOVA model for all genes in the dataset.  
```{r}
library(vegan)
library(factoextra)

scaled_test <-prcomp(test[c(1:24005)], scale=TRUE, center=TRUE)
fviz_eig(scaled_test)

# scale data
vegan <- scale(test[c(1:24005)])

# PerMANOVA 
permanova<-adonis2(vegan ~ parent*temperature, data = test, method='eu')
permanova
```

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegan ~ parent * temperature, data = test, method = "eu")
                   Df SumOfSqs      R2      F Pr(>F)    
parent              2   154649 0.12156 4.4308  0.001 ***
temperature         2   253981 0.19965 7.2767  0.001 ***
parent:temperature  4    78208 0.06148 1.1204  0.182    
Residual           45   785321 0.61731                  
Total              53  1272159 1.00000 

Gene expression is significantly different between temperature and parent, no interactive effect detected.  

# Examine PCA and sample distances of all genes  

Plot a PCA of samples by temperature and parental phenotype for all genes.        

```{r}
gPCAdata <- plotPCA(gvst, intgroup = c("temperature", "parent"), returnData=TRUE)
percentVar <- round(100*attr(gPCAdata, "percentVar")) #plot PCA of samples with all data

allgenesfilt_PCA <- ggplot(gPCAdata, aes(PC1, PC2, color=parent, shape=temperature)) + 
  geom_point(size=3) + 
  ggtitle(expression(bold("All genes (24,005 genes)")))+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  scale_color_manual(name="Parental Phenotype", values=c("Wildtype"="gray", "Bleached"="orange", "Nonbleached"="brown4"))+
  scale_shape_manual(name="Temperature", values=c(19,17,15))+
  geom_text(x=0, y=10, label="p[Parent]=0.001", color="black")+
  geom_text(x=0, y=8.5, label="p[Temperature]=0.001", color="black")+
  geom_text(x=0, y=7, label="p[Interaction]=0.182", color="darkgray")+
  theme_classic() + #Set background color
  theme(panel.border = element_blank(), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()); allgenesfilt_PCA

ggsave("figures/rna_seq/pca.pdf", allgenesfilt_PCA, width=8, height=4)
ggsave("figures/rna_seq/pca.jpeg", allgenesfilt_PCA, width=8, height=4)
```

Add a plot with elipses.  

```{r}
allgenes_ellipse<-allgenesfilt_PCA + stat_ellipse();allgenes_ellipse

ggsave("figures/rna_seq/pca_ellipse.pdf", allgenes_ellipse, width=9, height=5)
ggsave("figures/rna_seq/pca_ellipse.jpeg", allgenes_ellipse, width=9, height=5)
```

# Run DEG analysis of DEGs for all treatments  

Use the likelihood ratio approach based on guidance from DESeq2: "DESeq2 offers two kinds of hypothesis tests: the Wald test, where we use the estimated standard error of a log2 fold change to test if it is equal to zero, and the likelihood ratio test (LRT). The LRT examines two models for the counts, a full model with a certain number of terms and a reduced model, in which some of the terms of the full model are removed. The test determines if the increased likelihood of the data using the extra terms in the full model is more than expected if those extra terms are truly zero.

The LRT is therefore useful for testing multiple terms at once, for example testing 3 or more levels of a factor at once, or all interactions between two variables. The LRT for count data is conceptually similar to an analysis of variance (ANOVA) calculation in linear regression, except that in the case of the Negative Binomial GLM, we use an analysis of deviance (ANODEV), where the deviance captures the difference in likelihood between a full and a reduced model." 

Run differential expression test using LRT 
```{r, message = FALSE}
DEG <- DESeq(gdds, test="LRT", reduced=~1) #run differential expression test by temperature x parent using the LRT test 

DEG.results.all <- results(DEG)
resultsNames(DEG)
#this shows the comparisons made (shown similar to lm output in terms of interactions)
#The interaction terms temperature33.parentBleached and temperature33.parentNonbleached give the difference between the parental effect for a given temperature and the parental effect for the reference parent (Wildtype)

DEG.df<-as.data.frame(DEG.results.all)

DEG.results.all <- as.data.frame(subset(DEG.results.all, padj<0.01))
DEG.results.all <- as.data.frame(subset(DEG.results.all, abs(log2FoldChange)>1))
dim(DEG.results.all)

DEGlist <- gdds[rownames(DEG.results.all)]

DEGvst <- varianceStabilizingTransformation(DEGlist) #apply a variance stabilizing transforamtion to minimize effects of small counts and normalize
```
There are 935 genes differentially expressed at p<0.01 and log2FoldChange>1 between temperature x parent groups. Because we did not have interactive effects in the PERMANOVA, it makes sense that this is a lower number than within temperature and within parent comparisons that are done below. 

What does this value mean if we don't provide contrasts? 

View volcano plot. 
```{r}
#BiocManager::install('apeglm')
#BiocManager::install('EnhancedVolcano')

library(apeglm)
library(EnhancedVolcano)

res <- results(DEG)

volcano_plot<-EnhancedVolcano(res,
    lab = rownames(res),
    x = 'log2FoldChange',
    y = 'pvalue',
    title = '',
    pCutoff = 0.01,
    FCcutoff = 1,
    pointSize = 3.0,
    labSize = 0.0);volcano_plot

ggsave("figures/rna_seq/volcano.pdf", plot=volcano_plot, width=14, height=8)
ggsave("figures/rna_seq/volcano.png", plot=volcano_plot, width=14, height=8)
```

# View PCA of DEGs for all groups 

Conduct PERMANOVA by parent and temperature 

Export data for PERMANOVA test.  
```{r}
test_DEG<-t(assay(DEGvst)) #export as matrix
test_DEG<-as.data.frame(test_DEG)

#add category columns
test_DEG$sample<-rownames(test_DEG)
test_DEG$temperature<-metadata$temperature[match(test_DEG$sample, metadata$sample)]
test_DEG$parent<-metadata$parent[match(test_DEG$sample, metadata$sample)]
```

Build PERMANOVA model for DEG's between NB and B at 33°C.  
```{r}
dim(test_DEG)

scaled_test_DEG <-prcomp(test_DEG[c(1:700)], scale=TRUE, center=TRUE)
fviz_eig(scaled_test_DEG)

# scale data
vegan_DEG <- scale(test_DEG[c(1:700)])

# PerMANOVA 
permanova_DEG<-adonis2(vegan_DEG ~ parent*temperature, data = test_DEG, method='eu')
permanova_DEG
```

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegan_DEG ~ parent * temperature, data = test_DEG, method = "eu")
                   Df SumOfSqs      R2       F Pr(>F)    
parent              2     8747 0.23576 12.2931  0.001 ***
temperature         2     9526 0.25677 13.3886  0.001 ***
parent:temperature  4     2819 0.07597  1.9807  0.001 ***
Residual           45    16009 0.43151                   
Total              53    37100 1.00000  

There is an interaction, as expected because we are selecting DEG's.

Plot a PCA of samples by temperature and parental phenotype for DEGs.        
```{r}
gPCAdata_DEG <- plotPCA(DEGvst, intgroup = c("temperature", "parent"), returnData=TRUE)
percentVar_DEG <- round(100*attr(gPCAdata_DEG, "percentVar")) #plot PCA of samples with all data

DEG_PCA <- ggplot(gPCAdata_DEG, aes(PC1, PC2, color=parent, shape=temperature)) + 
  geom_point(size=3) + 
  ggtitle(expression(bold("DEGs: all groups (700 genes)")))+
  xlab(paste0("PC1: ",percentVar_DEG[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_DEG[2],"% variance")) +
  scale_color_manual(name="Parental Phenotype", values=c("Wildtype"="gray", "Bleached"="orange", "Nonbleached"="brown4"))+
  scale_shape_manual(name="Temperature", values=c(19,17,15))+
  geom_text(x=0, y=-2, label="p[Parent]=0.001", color="black")+
  geom_text(x=0, y=-3.5, label="p[Temperature]=0.001", color="black")+
  geom_text(x=0, y=-5, label="p[Interaction]=0.001", color="black")+
  theme_classic() + #Set background color
  theme(panel.border = element_blank(), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()); DEG_PCA

ggsave("figures/rna_seq/pca_DEG.pdf", DEG_PCA, width=8, height=4)
ggsave("figures/rna_seq/pca_DEG.jpeg", DEG_PCA, width=8, height=4)
```

# Add heatmap by parent and treatment  

```{r}
df <- as.data.frame(colData(gdds)[,c("temperature","parent")])
df$temperature <- factor(as.character(df$temperature), levels=c("27", "30", "33"))
df$parent <- factor(as.character(df$parent), levels=c("Wildtype", "Bleached", "Nonbleached"))
df<-df%>%
  arrange(temperature, parent)

library(viridis); library(pheatmap)

ann_colors = list(
    temperature = c("27" = "lightblue", "30" = "pink2","33"= "red2"), 
    parent = c("Bleached" = "orange", "Nonbleached" = "brown4", "Wildtype" = "gray")
)

legend_names_col = colnames(assay(DEGvst))

#pdf("figures/rna_seq/DEG_all_heatmap.pdf", width=15, height=15)
pheatmap(assay(DEGvst), cluster_rows=TRUE, show_rownames=FALSE, color=inferno(10), show_colnames=TRUE, fontsize_row=3, scale="row", cluster_cols=TRUE, annotation_col=df, annotation_colors = ann_colors, labels_col=legend_names_col)
#dev.off()
```

# Plot a heatmap of sample to sample distances    

```{r}
gsampleDists <- dist(t(assay(gvst))) #calculate distance matix
gsampleDistMatrix <- as.matrix(gsampleDists) #distance matrix
rownames(gsampleDistMatrix) <- colnames(gvst) #assign row names
colnames(gsampleDistMatrix) <- NULL #assign col names
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255) #assign colors

save_pheatmap_pdf <- function(x, filename, width=7, height=7) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}

pht<-pheatmap(gsampleDistMatrix, #plot matrix
         clustering_distance_rows=gsampleDists, #cluster rows
         clustering_distance_cols=gsampleDists, #cluster columns
         col=colors) #set colors

save_pheatmap_pdf(pht, "figures/rna_seq/sample_distances.pdf")
dev.off()
```

# DEG Analyses 

## Parent comparisons 

Specifically look for genes that are different between bleached and non bleached parental histories.  

### Nonbleached vs Bleached
```{r}
resultsNames(DEG)
parent_degNB_B<-results(DEG, contrast=c("parent","Nonbleached","Bleached"))
parent_degNB_B <- as.data.frame(subset(parent_degNB_B, padj<0.01))
parent_degNB_B <- as.data.frame(subset(parent_degNB_B, abs(log2FoldChange)>1))

dim(parent_degNB_B)
```
There are 1114 genes differentially expressed between nonbleached and bleached parents at p<0.01 and l2FC>1. 

```{r}
DEGlistNB_B <- gdds[rownames(parent_degNB_B)]
DEGvstNB_B <- varianceStabilizingTransformation(DEGlistNB_B) 

pdf("figures/rna_seq/DEG_heatmap_NBvsB.pdf", width=10, height=10)
pheatmap(assay(DEGvstNB_B), cluster_rows=TRUE, show_rownames=FALSE, color=inferno(10), show_colnames=TRUE, fontsize_row=3, scale="row", cluster_cols=TRUE, annotation_col=df, annotation_colors = ann_colors, labels_col=legend_names_col)
dev.off()
```

### Nonbleached vs Wildtype
```{r}
parent_parent_degNB_WT<-results(DEG, contrast=c("parent","Nonbleached","Wildtype"))
parent_parent_degNB_WT <- as.data.frame(subset(parent_parent_degNB_WT, padj<0.01))
parent_parent_degNB_WT <- as.data.frame(subset(parent_parent_degNB_WT, abs(log2FoldChange)>1))

dim(parent_parent_degNB_WT)
```
There are 855 genes differentially expressed between nonbleached and wildype parents at p<0.01 and l2FC>1. 

### Bleached vs Wildtype
```{r}
parent_parent_degB_WT<-results(DEG, contrast=c("parent","Bleached","Wildtype"))
parent_parent_degB_WT <- as.data.frame(subset(parent_parent_degB_WT, padj<0.01))
parent_parent_degB_WT <- as.data.frame(subset(parent_parent_degB_WT, abs(log2FoldChange)>1))

dim(parent_parent_degB_WT)
```
There are 1007 genes differentially expressed between wildtype and bleached parents at p<0.01 and l2FC>1. 

## Temperature comparisons 

Next, look at 27 vs 30 degrees and 27 vs 33 degrees. 

27 vs 30 degrees 
```{r}
temp30_deg<-results(DEG, contrast=c("temperature","30","27"))
temp30_deg <- as.data.frame(subset(temp30_deg, padj<0.01))
temp30_deg <- as.data.frame(subset(temp30_deg, abs(log2FoldChange)>1))

dim(temp30_deg)
```
There are 507 DEGs between 27 and 30 degrees. 

27 vs 33 degrees 
```{r}
temp33_deg<-results(DEG, contrast=c("temperature","33","27"))
temp33_deg <- as.data.frame(subset(temp33_deg, padj<0.01))
temp33_deg <- as.data.frame(subset(temp33_deg, abs(log2FoldChange)>1))

dim(temp33_deg)
```
There are 2533 DEGs between 27 and 33 degrees. 

```{r}
DEGlist27_33 <- gdds[rownames(temp33_deg)]
DEGvst27_33 <- varianceStabilizingTransformation(DEGlist27_33) 

pdf("figures/rna_seq/DEG_heatmap_27vs33.pdf", width=15, height=20)
pheatmap(assay(DEGvst27_33), cluster_rows=TRUE, show_rownames=FALSE, color=inferno(10), show_colnames=TRUE, fontsize_row=3, scale="row", cluster_cols=TRUE, annotation_col=df, annotation_colors = ann_colors, labels_col=legend_names_col)
dev.off()
```

Sample R93 looks like it has different expression from the others in the same group. It was not resequenced and has very high read depth (31+ million reads). Does not appear to be an outlier due to technical differences. No record of anything weird happening with this sample during collection in online or physical notebook. No recorded problems with extraction, concentrations of RNA were in mid to upper range. 

# Interaction betweeen NB and B at 33°C 

Examine the difference in NB and B larvae at 33°C. Analyze these contrasts to generate DEG's between B and NB at 33°C.  
```{r}
resultsNames(DEG)
parent_deg_int1<-results(DEG, contrast=list("temperature33.parentNonbleached", "temperature33.parentBleached"))
parent_deg_int1 <- as.data.frame(subset(parent_deg_int1, padj<0.01))
parent_deg_int1 <- as.data.frame(subset(parent_deg_int1, abs(log2FoldChange)>1))

dim(parent_deg_int1)
```
There are 941 genes differentially expressed genes in the interaction between temperature and parent. This shows DEG's contrasted between NB and B at 33°C. 

```{r}
DEGlist_int1 <- gdds[rownames(parent_deg_int1)]
DEGvst_int1 <- varianceStabilizingTransformation(DEGlist_int1) 

pdf("figures/rna_seq/DEG_heatmap_NBvsB_33C-Interaction.pdf", width=10, height=10)
pheatmap(assay(DEGvst_int1), cluster_rows=TRUE, show_rownames=FALSE, color=inferno(10), show_colnames=TRUE, fontsize_row=3, scale="row", cluster_cols=TRUE, annotation_col=df, annotation_colors = ann_colors, labels_col=legend_names_col)
dev.off()
```

## View PCA of DEGs between NB and B at 33°C 

Conduct PERMANOVA by parent and temperature 

Export data for PERMANOVA test.  
```{r}
test_int<-t(assay(DEGvst_int1)) #export as matrix
test_int<-as.data.frame(test_int)

#add category columns
test_int$sample<-rownames(test_int)
test_int$temperature<-metadata$temperature[match(test_int$sample, metadata$sample)]
test_int$parent<-metadata$parent[match(test_int$sample, metadata$sample)]
```

Build PERMANOVA model for DEG's between NB and B at 33°C.  
```{r}
dim(test_int)

scaled_test_int <-prcomp(test_int[c(1:941)], scale=TRUE, center=TRUE)
fviz_eig(scaled_test_int)

# scale data
vegan_int <- scale(test_int[c(1:941)])

# PerMANOVA 
permanova_int<-adonis2(vegan_int ~ parent*temperature, data = test_int, method='eu')
permanova_int
```

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegan_int ~ parent * temperature, data = test_int, method = "eu")
                   Df SumOfSqs      R2       F Pr(>F)    
parent              2    12824 0.25713 12.9333  0.001 ***
temperature         2    10376 0.20805 10.4647  0.001 ***
parent:temperature  4     4363 0.08749  2.2002  0.001 ***
Residual           45    22310 0.44733                   
Total              53    49873 1.00000   

There is an interaction, as expected because we are selecting DEG's that result from interactive effects. 

Plot a PCA of samples by temperature and parental phenotype for DEGs.        

```{r}
gPCAdata_int <- plotPCA(DEGvst_int1, intgroup = c("temperature", "parent"), returnData=TRUE)
percentVar_int <- round(100*attr(gPCAdata_int, "percentVar")) #plot PCA of samples with all data

int_PCA <- ggplot(gPCAdata_int, aes(PC1, PC2, color=parent, shape=temperature)) + 
  geom_point(size=3) + 
  ggtitle(expression(bold("DEGs: NB vs B at 33°C (941 genes)")))+
  xlab(paste0("PC1: ",percentVar_int[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_int[2],"% variance")) +
  scale_color_manual(name="Parental Phenotype", values=c("Wildtype"="gray", "Bleached"="orange", "Nonbleached"="brown4"))+
  scale_shape_manual(name="Temperature", values=c(19,17,15))+
  geom_text(x=0, y=-2, label="p[Parent]=0.001", color="black")+
  geom_text(x=0, y=-3.5, label="p[Temperature]=0.001", color="black")+
  geom_text(x=0, y=-5, label="p[Interaction]=0.001", color="black")+
  theme_classic() + #Set background color
  theme(panel.border = element_blank(), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()); int_PCA

ggsave("figures/rna_seq/pca_NBvsB_33°C.pdf", int_PCA, width=8, height=4)
ggsave("figures/rna_seq/pca_NBvsB_33°C.jpeg", int_PCA, width=8, height=4)
```

Larger degree of separation at 33°C compared to lower temperatures. 

# Genes unique to NB vs B only at 33°C

Find the list of DEGs in the interaction b/t NB and B at 30. Then find list of genes present in 30°C/27°C and keep the list only unique to 33°C. 

List of genes in NB vs B at 33°Ç
```{r}
resultsNames(DEG)
parent_deg_int3<-results(DEG, contrast=list("temperature33.parentNonbleached", "temperature33.parentBleached"))
parent_deg_int3 <- as.data.frame(subset(parent_deg_int3, padj<0.01))
parent_deg_int3 <- as.data.frame(subset(parent_deg_int3, abs(log2FoldChange)>1))

dim(parent_deg_int3)
```
There are 941 of these genes. 

List of genes in NB vs B at 30°Ç
```{r}
resultsNames(DEG)
parent_deg_int4<-results(DEG, contrast=list("parent_Bleached_vs_Wildtype", "parent_Nonbleached_vs_Wildtype"))
parent_deg_int4 <- as.data.frame(subset(parent_deg_int4, padj<0.01))
parent_deg_int4 <- as.data.frame(subset(parent_deg_int4, abs(log2FoldChange)>1))

dim(parent_deg_int4)
```
There are 1114 of these genes.

Now keep only the genes in the 33°C list that are not in the NB vs B list. 

```{r}
parent_deg_int3 <- rownames_to_column(parent_deg_int3, var = "row_names")
parent_deg_int4 <- rownames_to_column(parent_deg_int4, var = "row_names")
unique_genes <- anti_join(parent_deg_int3, parent_deg_int4, by = "row_names")
rownames(unique_genes) <- unique_genes$row_names
unique_genes$row_names <- NULL

dim(unique_genes)
str(unique_genes)
```
474

There are 474 genes unique to 33°C NB vs B comparisons compared to overall NB vs B comparisons.  

```{r}
DEGlist_unique <- gdds[rownames(unique_genes)]
DEGvst_unique <- varianceStabilizingTransformation(DEGlist_unique) 

pdf("figures/rna_seq/DEG_heatmap_NBvsB_33C-Unique.PDF", width=10, height=10)
pheatmap(assay(DEGvst_unique), cluster_rows=TRUE, show_rownames=FALSE, color=inferno(10), show_colnames=TRUE, fontsize_row=3, scale="row", cluster_cols=TRUE, annotation_col=df, annotation_colors = ann_colors, labels_col=legend_names_col)
dev.off()
```

## View PCA of DEGs between NB and B at 33°C 

Conduct PERMANOVA by parent and temperature 

Export data for PERMANOVA test.  
```{r}
test_unique<-t(assay(DEGvst_unique)) #export as matrix
test_unique<-as.data.frame(test_unique)

#add category columns
test_unique$sample<-rownames(test_unique)
test_unique$temperature<-metadata$temperature[match(test_unique$sample, metadata$sample)]
test_unique$parent<-metadata$parent[match(test_unique$sample, metadata$sample)]
```

Build PERMANOVA model for unique DEGs  
```{r}
dim(test_unique)

scaled_test_unique <-prcomp(test_unique[c(1:474)], scale=TRUE, center=TRUE)
fviz_eig(scaled_test_unique)

# scale data
vegan_unique <- scale(test_unique[c(1:474)])

# PerMANOVA 
permanova_unique<-adonis2(vegan_unique ~ parent*temperature, data = test_unique, method='eu')
permanova_unique
```

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegan_unique ~ parent * temperature, data = test_unique, method = "eu")
                   Df SumOfSqs      R2       F Pr(>F)    
parent              2   4381.6 0.17441  8.3935  0.001 ***
temperature         2   6320.8 0.25160 12.1083  0.001 ***
parent:temperature  4   2674.2 0.10645  2.5614  0.001 ***
Residual           45  11745.4 0.46754                   
Total              53  25122.0 1.00000 

There is an interaction, as expected because we are selecting DEG's that result from interactive effects. 

Plot a PCA of samples by temperature and parental phenotype for DEGs.        

```{r}
gPCAdata_unique <- plotPCA(DEGvst_unique, intgroup = c("temperature", "parent"), returnData=TRUE)
percentVar_unique <- round(100*attr(gPCAdata_unique, "percentVar")) #plot PCA of samples with all data

unique_PCA <- ggplot(gPCAdata_unique, aes(PC1, PC2, color=parent, shape=temperature)) + 
  geom_point(size=3) + 
  ggtitle(expression(bold("DEGs: Unique DEGs NB vs B at 33°C (474 genes)")))+
  xlab(paste0("PC1: ",percentVar_unique[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar_unique[2],"% variance")) +
  scale_color_manual(name="Parental Phenotype", values=c("Wildtype"="gray", "Bleached"="orange", "Nonbleached"="brown4"))+
  scale_shape_manual(name="Temperature", values=c(19,17,15))+
  geom_text(x=0, y=12, label="p[Parent]=0.001", color="black")+
  geom_text(x=0, y=10.5, label="p[Temperature]=0.001", color="black")+
  geom_text(x=0, y=9, label="p[Interaction]=0.001", color="black")+
  theme_classic() + #Set background color
  theme(panel.border = element_blank(), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()); unique_PCA

ggsave("figures/rna_seq/pca_NBvsB_33°C_unique.pdf", unique_PCA, width=8, height=4)
ggsave("figures/rna_seq/pca_NBvsB_33°C_unique.jpeg", unique_PCA, width=8, height=4)
```

# Export lists of genes for functional enrichment 

1. DEGs by parent
2. DEGs at temperature
3. DEGs at 33°C by B vs NB 
4. Unique DEGs only DEG at 33°C for B vs NB 



 

