---
title: "RNAseq DEG Functional Enrichment Analysis"
author: "Ariana S Huffmyer"
date: "4/21/2024"
output: html_document
editor_options: 
  chunk_output_type: console
---

1. read in files 
2. Split into up and down regulated for each contrast 
2. functional enrichment in goseq of specific contrasts or whole set and for up and down? Maybe run a loop for each analysis with "for" being for each contrast and within this each up vs down? 
3. Plot with contrast on X, GO on y, values as pvalue and fold change or num in cat? 
4. Output results table 

Repeat for temperature (27 vs 30, 27 vs 33, 30 vs 33) up vs down regulated 
Parent (NB vs WT, B vs WT, NB vs B) up vs down regulated 
Interaction (NB vs B at 30°C, NB vs B at 33°C) up vs down regulated 

# Set up 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# The following setting is important, do not omit.
options(stringsAsFactors = FALSE) #Set Strings to character
```

```{r}
if ("tidyverse" %in% rownames(installed.packages()) == 'FALSE') install.packages('tidyverse') 
if ("genefilter" %in% rownames(installed.packages()) == 'FALSE') install.packages('genefilter') 
if ("DESeq2" %in% rownames(installed.packages()) == 'FALSE') install.packages('DESeq2') 
if ("RColorBrewer" %in% rownames(installed.packages()) == 'FALSE') install.packages('RColorBrewer') 
if ("WGCNA" %in% rownames(installed.packages()) == 'FALSE') install.packages('WGCNA') 
if ("flashClust" %in% rownames(installed.packages()) == 'FALSE') install.packages('flashClust') 
if ("gridExtra" %in% rownames(installed.packages()) == 'FALSE') install.packages('gridExtra') 
if ("ComplexHeatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('ComplexHeatmap') 
if ("goseq" %in% rownames(installed.packages()) == 'FALSE') install.packages('goseq') 
if ("dplyr" %in% rownames(installed.packages()) == 'FALSE') install.packages('dplyr') 
if ("clusterProfiler" %in% rownames(installed.packages()) == 'FALSE') install.packages('clusterProfiler') 
if ("pheatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('pheatmap') 
if ("magrittr" %in% rownames(installed.packages()) == 'FALSE') install.packages('magrittr') 
if ("rtracklayer" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rtracklayer")
if ("GenomicRanges" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GenomicRanges")
if ("plyranges" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("plyranges")
if ("GSEABase" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GSEABase")
if ("GOSim" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GOSim")
if ("stats" %in% rownames(installed.packages()) == 'FALSE') install.packages("stats")
if ("ggdendro" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("ggdendro")
if ("GO.db" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GO.db")
if ("rrvgo" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rrvgo")

#BiocManager::install("org.Ce.eg.db", force=TRUE) #install if needed 

library("tidyverse")
library("genefilter")
library("DESeq2")
library("RColorBrewer")
library("WGCNA")
library("flashClust")
library("gridExtra")
library("ComplexHeatmap")
library("goseq")
library("dplyr")
library("clusterProfiler")
library("pheatmap")
library("magrittr")
library("rtracklayer")
library("GenomicRanges")
library("plyranges")
library("GSEABase")
library("GOSim")
library("stats")
library("ggdendro")
library("GO.db")
library("rrvgo")
```

# Read in data files 

1. Read in file with vst transformed counts of all genes detected and kept after filtering. 
```{r}
all_genes<-read.csv("output/rna_seq/all_genes.csv")
```

2. Read in file of DEG's between parental phenotypes
```{r}
temperature_DEG_list<-read.csv("output/rna_seq/temperature_deg_list.csv")
temperature_DEG_counts<-read.csv("output/rna_seq/temperature_deg_counts.csv")

temperature_DEG_list$contrast<-factor(temperature_DEG_list$contrast, levels=c("30vs27", "33vs27", "33vs30"))
```

3. Read in file of DEG's between temperatures 
```{r}
parent_DEG_list<-read.csv("output/rna_seq/parent_deg_list.csv")
parent_DEG_counts<-read.csv("output/rna_seq/parent_deg_counts.csv")
```

4. Read in file of DEG's between NB and B at 30-33°C 
```{r}
interaction_DEG_list<-read.csv("output/rna_seq/NBvsB_30-33_deg_list.csv")
interaction_DEG_counts<-read.csv("output/rna_seq/NBvsB_30-33_deg_counts.csv")
```

The list dataframes include DESeq2 results for each gene identified as a DEG in each contrast.  

The counts dataframes include gene counts for each gene in the DEG list for each sample. The number of genes may be lower than the number in the full list, because some genes are shared between contrasts in the list.  

# Read in annotation files 

Downloaded functional annotation (version 3) from: http://cyanophora.rutgers.edu/montipora/Montipora_capitata_HIv3.genes.EggNog_results.txt.gz (EggNog) and http://cyanophora.rutgers.edu/montipora/Montipora_capitata_HIv3.genes.KEGG_results.txt.gz (KEGG). 

Scripts to generate this file are from Mcapitata 2020 project scripts here: https://github.com/AHuffmyer/EarlyLifeHistory_Energetics 

Downloaded on 21 April 2024. Unzipped on desktop. I then removed the # in #query in the first column, otherwise it does not read in column names.    

```{r}
Mcap.annot <- read.table("data/rna_seq/Montipora_capitata_HIv3.genes.EggNog_results.txt",  quote="", sep="\t", header=TRUE)
dim(Mcap.annot)

# column names
head(Mcap.annot)
tail(Mcap.annot)

Mcap.annot<-Mcap.annot%>%
  rename(gene=query)
```

This functional annotation as 24,072 genes. This is 44% of total protein predicted genes described in the publication associated with this annotation. https://academic.oup.com/gigascience/article/doi/10.1093/gigascience/giac098/6815755 

We had 24,005 genes in our dataset that were detected and present at counts >10. 

Match up genes in gene list file to annotation file
```{r}
names(Mcap.annot)

#remove sample column name 
probes = names(all_genes)
#probes <- gsub("sample", "", probes)
#probes <- probes[probes != ""]

probes2annot = match(probes, Mcap.annot$gene) #match genes in datExpr to genes in annotation file, note I removed the # before query in this file before loading

# The following is the number of probes without annotation 
sum(is.na(probes2annot))

row_nas<-which(is.na(probes2annot))

#view the genes that do not have a match in the annotation file
missing<-as.data.frame(probes[row_nas])
print(missing)
```

There are 9390 genes from our data set that are not in the annotation file but that are in the genome. 

## Generate a vector of gene lenghts 

```{r}
#import file
gff <- rtracklayer::import("data/rna_seq/Montipora_capitata_HIv3.genes_fixed.gff3") #if this doesn't work, restart R and try again 

transcripts <- subset(gff, type == "transcript") #keep only transcripts 

transcripts_gr <- makeGRangesFromDataFrame(transcripts, keep.extra.columns=TRUE) #extract length information

transcript_lengths <- width(transcripts_gr) #isolate length of each transcript

seqnames<-transcripts_gr$ID #extract list of gene id 

lengths<-cbind(seqnames, transcript_lengths)

lengths<-as.data.frame(lengths) #convert to data frame
```

Add in length to the annotation file.    

```{r}
Mcap.annot$length<-lengths$transcript_lengths[match(Mcap.annot$gene, lengths$seqnames)]

which(is.na(Mcap.annot$length)) #all genes have lengths 
```

Format GO terms to remove dashes and quotes.  
```{r}
#remove dashes, remove quotes from columns and separate by semicolons (replace , with ;) in KEGG_ko, KEGG_pathway, and Annotation.GO.ID columns
head(Mcap.annot$GOs)
Mcap.annot$GOs <- gsub(",", ";", Mcap.annot$GOs)
Mcap.annot$GOs <- gsub('"', "", Mcap.annot$GOs)
Mcap.annot$GOs <- gsub("-", NA, Mcap.annot$GOs)
head(Mcap.annot$GOs)
```

# Temperature DEGs 

## Add in annotation information

Merge in relevant annotation information to the DEG list file.  

```{r}
temperature_DEG_list<-left_join(temperature_DEG_list, Mcap.annot, by="gene")
```

## Plot fold change of temperature DEGs

```{r}
temp_deg_FC <- temperature_DEG_list %>% 
  dplyr::arrange(log2FoldChange) %>%
  
  ggplot(aes(x=log2FoldChange, y=reorder(gene, -log2FoldChange), colour=padj)) +
  facet_grid(~contrast, scales="free_y")+
  geom_point(size=2)+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  geom_segment(aes(x=0, xend=log2FoldChange, y=gene, yend=gene),linewidth=1) +
  geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("Gene")+
  theme(legend.position = "right",
        axis.text.y = element_blank(),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Fold Change: Temperature DEGs'); temp_deg_FC

ggsave(plot=temp_deg_FC, "figures/rna_seq/functional_enrichment/temperature_DEG_foldchange.png", height=12, width=12)
```

## Prepare up and down regulated dataframes 

```{r}
dim(temperature_DEG_list)

temperature_up_list<-temperature_DEG_list%>%
  filter(log2FoldChange>0)

dim(temperature_up_list)

temperature_down_list<-temperature_DEG_list%>%
  filter(log2FoldChange<0)

dim(temperature_down_list)
```
1182 total genes with 518 up regulated and 664 down regulated. We are keeping all contrasts in these lists and we will run a loop to conduct enrichment for each contrast.    

## Conduct enrichment for up regulated genes 

Make a list of contrasts to loop through.  

```{r}
levels(temperature_up_list$contrast)
temp_contrasts<-c("30vs27", "33vs27", "33vs30")
```

Conduct functional enrichment for each contrast using goseq and rrvgo. Compare against all genes detected in the dataset. I am choosing to compare against genes in our dataset because we are looking at one developmental stage (larvae). Therefore, I do not want to conduct enrichment of all genes in the genome because many functions including calcification and gametogenesis do not apply to the biological function of larvae.  

Conduct at biological process level. Filter for adjusted p-value <0.01 because we have a lot of DEGs.  
```{r}
for (temp in temp_contrasts) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- temperature_up_list%>%
      filter(contrast==temp)%>%
      pull(gene)
    
    ##Get a list of GO Terms for each constrast
    GO.terms <- temperature_up_list%>%
      filter(contrast==temp)%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.01) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("output/rna_seq/functional_enrichment/goseq_temperature_up_", temp, ".csv"))
    
}
```

Create an empty dataframe to populate (only do this once).  

```{r}
temp_go_results <- data.frame()
```

Run a loop to filter enrichment for each temperature contrast using rrvgo. 
```{r}
temp_contrasts<-c("30vs27", "33vs27", "33vs30")

for (temp in temp_contrasts) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("output/rna_seq/functional_enrichment/goseq_temperature_up_", temp, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>5)%>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$direction<-c("upregulated")
    go_results$contrast<-temp
    
    #add to data frame 
    temp_go_df <- go_results
    temp_go_results <- rbind(temp_go_results, temp_go_df) 

}
```

```{r}
dim(temp_go_results)
```
702 total enriched GO terms.  

## Conduct enrichment for down regulated genes 

Make a list of contrasts to loop through.  

```{r}
temp_contrasts<-c("30vs27", "33vs27", "33vs30")
```

Conduct at biological process level. Filter for adjusted p-value <0.01 because we have a lot of DEGs.  
```{r}
for (temp in temp_contrasts) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- temperature_down_list%>%
      filter(contrast==temp)%>%
      pull(gene)
    
    ##Get a list of GO Terms for each constrast
    GO.terms <- temperature_down_list%>%
      filter(contrast==temp)%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.01) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("output/rna_seq/functional_enrichment/goseq_temperature_down_", temp, ".csv"))
    
}
```

Run a loop to filter enrichment for each temperature contrast using rrvgo.   
```{r}
temp_contrasts<-c("30vs27", "33vs27", "33vs30")

for (temp in temp_contrasts) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("output/rna_seq/functional_enrichment/goseq_temperature_down_", temp, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      #filter(numInCat>5)%>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$direction<-c("downregulated")
    go_results$contrast<-temp
    
    #add to data frame 
    temp_go_df <- go_results
    temp_go_results <- rbind(temp_go_results, temp_go_df) 

}
```

```{r}
dim(temp_go_results)
```
There are now 3,651 enriched GO terms (702 were upregulated, leaving 2,949 downregulated). 

## Plot enrichment for up and down regulated genes for each contrast 

Upregulated genes 
```{r}
temp_go_plot_up <- temp_go_results %>% 
  filter(direction=="upregulated")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  
  ggplot(aes(x=contrast, y=GOterm, colour=over_represented_pvalue)) +
  facet_grid(ParentTerm ~ ., scales="free", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=numDEInCat))+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  scale_size(name="numDEInCat", trans="reverse", range = c(0.1, 2))+
  #geom_segment(aes(x=0, xend=-log10(over_represented_pvalue), y=GOterm, yend=GOterm),linewidth=1) +
  #geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("GO Term")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Enrichment: Upregulated Temperature DEGs'); temp_go_plot_up

ggsave(plot=temp_go_plot_up, "figures/rna_seq/functional_enrichment/temperature_GO_upregulated.png", height=80, width=6, limitsize=FALSE)
```

Downregulated genes 
```{r}
temp_go_plot_down <- temp_go_results %>% 
  filter(direction=="downregulated")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  
  ggplot(aes(x=contrast, y=GOterm, colour=over_represented_pvalue)) +
  facet_grid(ParentTerm ~ ., scales="free", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=numDEInCat))+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  scale_size(name="numDEInCat", trans="reverse", range = c(0.1, 2))+
  #geom_segment(aes(x=0, xend=-log10(over_represented_pvalue), y=GOterm, yend=GOterm),linewidth=1) +
  #geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("GO Term")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Enrichment: Downregulated Temperature DEGs'); temp_go_plot_down

ggsave(plot=temp_go_plot_down, "figures/rna_seq/functional_enrichment/temperature_GO_downregulated.png", height=100, width=6, limitsize=FALSE)
```

There are so many terms its hard to use these! Will need to return to this.  

# Parent DEGs 

## Add in annotation information

Merge in relevant annotation information to the DEG list file.  

```{r}
parent_DEG_list<-left_join(parent_DEG_list, Mcap.annot, by="gene")
```

## Plot fold change of temperature DEGs

```{r}
parent_deg_FC <- parent_DEG_list %>% 
  dplyr::arrange(log2FoldChange) %>%
  
  ggplot(aes(x=log2FoldChange, y=reorder(gene, -log2FoldChange), colour=padj)) +
  facet_grid(~contrast, scales="free_y")+
  geom_point(size=2)+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  geom_segment(aes(x=0, xend=log2FoldChange, y=gene, yend=gene),linewidth=1) +
  geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("Gene")+
  theme(legend.position = "right",
        axis.text.y = element_blank(),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Fold Change: Parent DEGs'); parent_deg_FC

ggsave(plot=parent_deg_FC, "figures/rna_seq/functional_enrichment/parent_DEG_foldchange.png", height=12, width=12)
```

## Prepare up and down regulated dataframes 

```{r}
dim(parent_DEG_list)

parent_up_list<-parent_DEG_list%>%
  filter(log2FoldChange>0)

dim(parent_up_list)

parent_down_list<-parent_DEG_list%>%
  filter(log2FoldChange<0)

dim(parent_down_list)
```
601 total genes with 236 up regulated and 365 down regulated. We are keeping all contrasts in these lists and we will run a loop to conduct enrichment for each contrast.    

## Conduct enrichment for up regulated genes 

Make a list of contrasts to loop through.  

```{r}
levels(as.factor(parent_up_list$contrast))
#parent_contrasts<-c("BvsWT", "NBvsWT", "NBvsB")
parent_contrasts<-c("BvsWT", "NBvsB")
```
Note that NB vs WT genes (27) had no GO terms - all genes had GO terms of "NA" and therefore I couldn't do enrichment.  

Conduct functional enrichment for each contrast using goseq and rrvgo. Compare against all genes detected in the dataset. I am choosing to compare against genes in our dataset because we are looking at one developmental stage (larvae). Therefore, I do not want to conduct enrichment of all genes in the genome because many functions including calcification and gametogenesis do not apply to the biological function of larvae.  

Conduct at biological process level. Filter for adjusted p-value <0.01 because we have a lot of DEGs.  
```{r}
for (parent in parent_contrasts) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- parent_up_list%>%
      filter(contrast==parent)%>%
      pull(gene)
    
    ##Get a list of GO Terms for each constrast
    GO.terms <- parent_up_list%>%
      filter(contrast==parent)%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.01) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("output/rna_seq/functional_enrichment/goseq_parent_up_", parent, ".csv"))
    
}
```

Create an empty dataframe to populate (only do this once).  

```{r}
parent_go_results <- data.frame()
```

Run a loop to filter enrichment for each temperature contrast using rrvgo. 
```{r}
parent_contrasts<-c("BvsWT", "NBvsB")

for (parent in parent_contrasts) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("output/rna_seq/functional_enrichment/goseq_parent_up_", parent, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>1)%>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$direction<-c("upregulated")
    go_results$contrast<-parent
    
    #add to data frame 
    parent_go_df <- go_results
    parent_go_results <- rbind(parent_go_results, parent_go_df) 

}
```

```{r}
dim(parent_go_results)
```
276 total enriched GO terms.  

There won't be NB vs WT comparison here, there are no enriched GO terms for this comparison that are upregulated.  

## Conduct enrichment for down regulated genes 

Make a list of contrasts to loop through.  

```{r}
parent_contrasts<-c("BvsWT", "NBvsWT", "NBvsB")
```

Conduct at biological process level. Filter for adjusted p-value <0.01 because we have a lot of DEGs.  
```{r}
for (parent in parent_contrasts) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- parent_down_list%>%
      filter(contrast==parent)%>%
      pull(gene)
    
    ##Get a list of GO Terms for each constrast
    GO.terms <- parent_down_list%>%
      filter(contrast==parent)%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.01) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("output/rna_seq/functional_enrichment/goseq_parent_down_", parent, ".csv"))
    
}
```

Run a loop to filter enrichment for each temperature contrast using rrvgo.   
```{r}
parent_contrasts<-c("BvsWT", "NBvsWT", "NBvsB")

for (parent in parent_contrasts) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("output/rna_seq/functional_enrichment/goseq_parent_down_", parent, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>1)%>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$direction<-c("downregulated")
    go_results$contrast<-parent
    
    #add to data frame 
    parent_go_df <- go_results
    parent_go_results <- rbind(parent_go_results, parent_go_df) 

}
```

```{r}
dim(parent_go_results)
```
There are now 1,375 enriched GO terms (276 were upregulated, leaving 1,099 downregulated). 

## Plot enrichment for up and down regulated genes for each contrast 

Upregulated genes 
```{r}
parent_go_plot_up <- parent_go_results %>% 
  filter(direction=="upregulated")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  
  ggplot(aes(x=contrast, y=GOterm, colour=over_represented_pvalue)) +
  facet_grid(ParentTerm ~ ., scales="free", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=numDEInCat))+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  scale_size(name="numDEInCat", trans="reverse", range = c(0.1, 2))+
  scale_x_discrete(limits=c("BvsWT", "NBvsWT", "NBvsB"))+
  #geom_segment(aes(x=0, xend=-log10(over_represented_pvalue), y=GOterm, yend=GOterm),linewidth=1) +
  #geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("GO Term")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Enrichment: Upregulated Parent DEGs'); parent_go_plot_up

ggsave(plot=parent_go_plot_up, "figures/rna_seq/functional_enrichment/parent_GO_upregulated.png", height=40, width=6, limitsize=FALSE)
```

Downregulated genes 
```{r}
parent_go_plot_down <- parent_go_results %>% 
  filter(direction=="downregulated")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  
  ggplot(aes(x=contrast, y=GOterm, colour=over_represented_pvalue)) +
  facet_grid(ParentTerm ~ ., scales="free", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=numDEInCat))+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  scale_size(name="numDEInCat", trans="reverse", range = c(0.1, 2))+
  #geom_segment(aes(x=0, xend=-log10(over_represented_pvalue), y=GOterm, yend=GOterm),linewidth=1) +
  #geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("GO Term")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Enrichment: Downregulated Parent DEGs'); parent_go_plot_down

ggsave(plot=parent_go_plot_down, "figures/rna_seq/functional_enrichment/parent_GO_downregulated.png", height=80, width=6, limitsize=FALSE)
```

There are so many terms its hard to use these! Will need to return to this.  

# Interaction DEGs 

DEGs between parent at elevated temperature. 

## Add in annotation information

Merge in relevant annotation information to the DEG list file.  

```{r}
interaction_DEG_list<-left_join(interaction_DEG_list, Mcap.annot, by="gene")
```

## Plot fold change of temperature DEGs

```{r}
interaction_deg_FC <- interaction_DEG_list %>%
  mutate(label=paste0(Description, "-", gene))%>%
  dplyr::arrange(log2FoldChange) %>%
  
  ggplot(aes(x=log2FoldChange, y=reorder(label, -log2FoldChange), colour=padj)) +
  facet_grid(~contrast, scales="free_y")+
  geom_point(size=2)+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  geom_segment(aes(x=0, xend=log2FoldChange, y=label, yend=label),linewidth=1) +
  geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("Description - Gene")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Fold Change: Parent DEGs at 30-33°C'); interaction_deg_FC

ggsave(plot=interaction_deg_FC, "figures/rna_seq/functional_enrichment/interaction_DEG_foldchange.png", height=12, width=18)
```

## Prepare up and down regulated dataframes 

```{r}
dim(interaction_DEG_list)

interaction_up_list<-interaction_DEG_list%>%
  filter(log2FoldChange>0)

dim(interaction_up_list)

interaction_down_list<-interaction_DEG_list%>%
  filter(log2FoldChange<0)

dim(interaction_down_list)
```
56 total genes with 30 up regulated and 26 down regulated. We are keeping all contrasts in these lists and we will run a loop to conduct enrichment for each contrast.    

We may not have enough genes for enrichment and may instead chose to do more manual investigation.  

## Conduct enrichment for up regulated genes 

Make a list of contrasts to loop through.  

```{r}
levels(as.factor(interaction_up_list$contrast))
interaction_contrasts<-c("NBvsB_30C", "NBvsB_33C")
```
Conduct functional enrichment for each contrast using goseq and rrvgo. Compare against all genes detected in the dataset. I am choosing to compare against genes in our dataset because we are looking at one developmental stage (larvae). Therefore, I do not want to conduct enrichment of all genes in the genome because many functions including calcification and gametogenesis do not apply to the biological function of larvae.  

Conduct at biological process level. Filter for adjusted p-value <0.01 because we have a lot of DEGs.  
```{r}
for (interaction in interaction_contrasts) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- interaction_up_list%>%
      filter(contrast==interaction)%>%
      pull(gene)
    
    ##Get a list of GO Terms for each constrast
    GO.terms <- interaction_up_list%>%
      filter(contrast==interaction)%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.01) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("output/rna_seq/functional_enrichment/goseq_interaction_up_", interaction, ".csv"))
    
}
```

Create an empty dataframe to populate (only do this once).  

```{r}
interaction_go_results <- data.frame()
```

Run a loop to filter enrichment for each temperature contrast using rrvgo. 
```{r}
interaction_contrasts<-c("NBvsB_30C", "NBvsB_33C")

for (interaction in interaction_contrasts) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("output/rna_seq/functional_enrichment/goseq_interaction_up_", interaction, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      #filter(numInCat>1)%>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$direction<-c("upregulated")
    go_results$contrast<-interaction
    
    #add to data frame 
    interaction_go_df <- go_results
    interaction_go_results <- rbind(interaction_go_results, interaction_go_df) 

}
```

```{r}
dim(interaction_go_results)
```
281 total enriched GO terms.  

## Conduct enrichment for down regulated genes 

Make a list of contrasts to loop through.  

```{r}
interaction_contrasts<-c("NBvsB_30C", "NBvsB_33C")
```

Conduct at biological process level. Filter for adjusted p-value <0.01 because we have a lot of DEGs.  
```{r}
for (interaction in interaction_contrasts) {
  
    ### Generate vector with names of all genes detected in our dataset 
    ALL.vector <- c(Mcap.annot$gene)
    
    ### Generate length vector for all genes 
    LENGTH.vector <- as.integer(Mcap.annot$length)
    
    ### Generate vector with names in just the contrast we are analyzing
    ID.vector <- interaction_down_list%>%
      filter(contrast==interaction)%>%
      pull(gene)
    
    ##Get a list of GO Terms for each constrast
    GO.terms <- interaction_down_list%>%
      filter(contrast==interaction)%>%
      dplyr::select(gene, GOs)
    
    ##Format to have one goterm per row with gene ID repeated
    split <- strsplit(as.character(GO.terms$GOs), ";") 
    split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
    colnames(split2) <- c("gene", "GOs")
    GO.terms<-split2

    ##Construct list of genes with 1 for genes in contrast and 0 for genes not in the contrast
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names

    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP"), method="Wallenius", use_genes_without_cat=TRUE)

    GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
    colnames(GO)[1] <- "GOterm"
    
    #adjust p-values 
    GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    GO <- GO %>%
        dplyr::filter(bh_adjust<0.01) %>%
        dplyr::arrange(., ontology, bh_adjust)

    #Write file of results 
    write_csv(GO, file = paste0("output/rna_seq/functional_enrichment/goseq_interaction_down_", interaction, ".csv"))
    
}
```

Run a loop to filter enrichment for each temperature contrast using rrvgo.   
```{r}
interaction_contrasts<-c("NBvsB_30C", "NBvsB_33C")

for (interaction in interaction_contrasts) {

    #Read relevant file of results from goseq analysis  
    go_results<-read_csv(file = paste0("output/rna_seq/functional_enrichment/goseq_interaction_down_", interaction, ".csv"))
    
    go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      #filter(numInCat>1)%>%
      arrange(., bh_adjust)
    
    #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=c("BP"),
                                method="Rel")
    #claculate similarity 
    scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    #add in contrast and regulation
    go_results$direction<-c("downregulated")
    go_results$contrast<-interaction
    
    #add to data frame 
    interaction_go_df <- go_results
    interaction_go_results <- rbind(interaction_go_results, interaction_go_df) 

}
```

```{r}
dim(interaction_go_results)
```
There are now 793 enriched GO terms (281 were upregulated, leaving 512 downregulated). 

## Plot enrichment for up and down regulated genes for each contrast 

Upregulated genes 
```{r}
interaction_go_plot_up <- interaction_go_results %>% 
  filter(direction=="upregulated")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  
  ggplot(aes(x=contrast, y=GOterm, colour=over_represented_pvalue)) +
  facet_grid(ParentTerm ~ ., scales="free", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=numDEInCat))+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  scale_size(name="numDEInCat", trans="reverse", range = c(0.1, 2))+
  #scale_x_discrete(limits=c("BvsWT", "NBvsWT", "NBvsB"))+
  #geom_segment(aes(x=0, xend=-log10(over_represented_pvalue), y=GOterm, yend=GOterm),linewidth=1) +
  #geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("GO Term")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Enrichment: Upregulated Parent DEGs at 30-33°C'); interaction_go_plot_up

ggsave(plot=interaction_go_plot_up, "figures/rna_seq/functional_enrichment/interaction_GO_upregulated.png", height=40, width=6, limitsize=FALSE)
```

Downregulated genes 
```{r}
interaction_go_plot_down <- interaction_go_results %>% 
  filter(direction=="downregulated")%>%
  dplyr::arrange(over_represented_pvalue) %>%
  
  ggplot(aes(x=contrast, y=GOterm, colour=over_represented_pvalue)) +
  facet_grid(ParentTerm ~ ., scales="free", labeller = label_wrap_gen(width = 2, multi_line = TRUE))+
  geom_point(aes(size=numDEInCat))+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  scale_size(name="numDEInCat", trans="reverse", range = c(0.1, 2))+
  #geom_segment(aes(x=0, xend=-log10(over_represented_pvalue), y=GOterm, yend=GOterm),linewidth=1) +
  #geom_vline(xintercept = 0, linetype = "dashed") + # Add vertical line
  theme_classic() +
  ylab("GO Term")+
  theme(legend.position = "right",
        axis.text.y = element_text(size=8),
        axis.title.y = element_text(size=12, face="bold"),
        axis.title.x = element_text(size = 12, face = "bold"),
        plot.title = element_text(size = 14, face = "bold"))+
  ggtitle('Enrichment: Downregulated Parent DEGs at 30-33°C'); interaction_go_plot_down

ggsave(plot=interaction_go_plot_down, "figures/rna_seq/functional_enrichment/interaction_GO_downregulated.png", height=80, width=6, limitsize=FALSE)
```

There are so many terms its hard to use these! Will need to return to this. 


We dont have many genes in this list, but lots of enriched GO terms - why? 

Plot interaction gene list with description/family information 




