---
title: "RNAseq Differential Gene Expression Analysis"
author: "Ariana S Huffmyer"
date: "2/18/2022"
output: html_document
editor_options: 
  chunk_output_type: console
---

Analysis of RNAseq Differential Gene Expression (DEG).  

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# The following setting is important, do not omit.
options(stringsAsFactors = FALSE) #Set Strings to character
```

Load required libraries.  

```{r}
if ("tidyverse" %in% rownames(installed.packages()) == 'FALSE') install.packages('tidyverse') 
if ("genefilter" %in% rownames(installed.packages()) == 'FALSE') install.packages('genefilter') 
if ("DESeq2" %in% rownames(installed.packages()) == 'FALSE') install.packages('DESeq2') 
if ("RColorBrewer" %in% rownames(installed.packages()) == 'FALSE') install.packages('RColorBrewer') 
if ("WGCNA" %in% rownames(installed.packages()) == 'FALSE') install.packages('WGCNA') 
if ("flashClust" %in% rownames(installed.packages()) == 'FALSE') install.packages('flashClust') 
if ("gridExtra" %in% rownames(installed.packages()) == 'FALSE') install.packages('gridExtra') 
if ("ComplexHeatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('ComplexHeatmap') 
if ("goseq" %in% rownames(installed.packages()) == 'FALSE') install.packages('goseq') 
if ("dplyr" %in% rownames(installed.packages()) == 'FALSE') install.packages('dplyr') 
if ("clusterProfiler" %in% rownames(installed.packages()) == 'FALSE') install.packages('clusterProfiler') 
if ("pheatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('pheatmap') 
if ("magrittr" %in% rownames(installed.packages()) == 'FALSE') install.packages('magrittr') 
if ("vegan" %in% rownames(installed.packages()) == 'FALSE') install.packages('vegan') 

library("tidyverse")
library("genefilter")
library("WGCNA")
library("DESeq2")
library("RColorBrewer")
library("flashClust")
library("gridExtra")
library("goseq")
library("dplyr")
library("clusterProfiler")
library("pheatmap")
library("magrittr")
library("vegan")
```

# **Data input and filtering**  

Import the data files.  

```{r}
#load metadata sheet with sample name and developmental stage information
metadata <- read.csv("data/rna_seq/sample_rnaseq_metadata.csv", header = TRUE, sep = ",")%>%select(sample, temperature, parent, symbiont, phenotype)
metadata$code<-paste(metadata$temperature, "-", metadata$parent)
head(metadata)

#load gene count matrix generated from cluster computation
gcount <- as.data.frame(read.csv("data/rna_seq/Mcapitata2023_gene_count_matrix.csv", row.names="gene_id"), colClasses = double)
head(gcount)
```

Check that there are no genes with 0 counts across all samples. 

```{r}
dim(gcount) 

gcount<-gcount %>%
     mutate(Total = rowSums(.[, 1:54]))%>%
    filter(!Total==0)%>%
    select(!Total)

dim(gcount)
```

We started with 54,384 genes. which is the total number in the annotation. About 15,000 genes that had 0's across all samples (not detected) with 39,498 remaining.    

Conduct data filtering, this includes:  

*pOverA*: Specifying the minimum count for a proportion of samples for each gene. Here, we are using a pOverA of 0.05. This is because we have 54 samples with a minimum of n=6 samples  per group. Therefore, we will accept genes that are present in 6/54 = 0.11 of the samples because we expect different expression by coral colony We are further setting the minimum count of genes to 10, such that 11% of the samples must have a gene count of >10 in order for the gene to remain in the data set.  

Filter in the package "genefilter". Pre-filtering our dataset to reduce the memory size dataframe, increase the speed of the transformation and testing functions, and improve quality of statistical analysis by removing low-coverage counts. Removed counts could represent outliers in the data and removing these improves sensitivity of statistical tests.

```{r}
filt <- filterfun(pOverA(0.1,11))

#create filter for the counts data
gfilt <- genefilter(gcount, filt)

#identify genes to keep by count filter
gkeep <- gcount[gfilt,]

#identify gene lists
gn.keep <- rownames(gkeep)

#gene count data filtered in PoverA, P percent of the samples have counts over A
gcount_filt <- as.data.frame(gcount[which(rownames(gcount) %in% gn.keep),])

#How many rows do we have before and after filtering?
nrow(gcount) #Before
nrow(gcount_filt) #After
```

Before filtering, we had approximately 39,498 genes. After filtering for pOverA, we have approximately 24,003 genes. This indicates that there were about 24,003 genes present in 10% of samples at <10 counts per gene.  

Prepare sample names. Keep R and any numbers following R in the colnames of gcount_filt.  

```{r}
# Function to extract 'R' followed by numbers from column names
extract_R_numbers <- function(col_names) {
  sub("R(\\d+).*", "\\1", col_names)
}

# Apply the function to all column names
new_col_names <- sapply(names(gcount_filt), extract_R_numbers)

# Rename the columns
names(gcount_filt) <- paste0("R", new_col_names)

# Output the modified dataframe
names(gcount_filt)
length(names(gcount_filt))
```

Column names are now correct and there are 54 as expected.  

Print the sample names in the metadata file. 

```{r}
metadata$sample
```

In order for the DESeq2 algorithms to work, the SampleIDs on the metadata file and count matrices have to match exactly and in the same order. They are.  

# **Format metadata and count data columns**  

Convert temperature and metadata categories to factors. 
```{r}
metadata$temperature<-as.factor(metadata$temperature)
metadata$parent<-as.factor(metadata$parent)
metadata$symbiont<-as.factor(metadata$symbiont)
metadata$code<-as.factor(metadata$code)
```

Make sure the metadata and the columns in the gene count matrix are all the same.  
```{r}
metadata$sample
colnames(gcount_filt)
```

Reorder them automatically if needed.  
```{r}
list<-colnames(gcount_filt)
list<-as.factor(list)

metadata$sample<-as.factor(metadata$sample)

# Re-order the levels
metadata$sample <- factor(as.character(metadata$sample), levels=list)
# Re-order the data.frame
metadata_ordered <- metadata[order(metadata$sample),]
metadata_ordered$sample

head(metadata_ordered)

metadata_ordered$sample
colnames(gcount_filt)
head(gcount_filt)
```

These match we are good to continue to the next step!    

# **Construct DESeq2 data set by temperature x parent**  

Create a DESeqDataSet design from gene count matrix and labels. Here we set the design to look at temperature to test for any differences in gene expression across temperatures  
```{r}
#Set DESeq2 design
gdds <- DESeqDataSetFromMatrix(countData = gcount_filt,
                              colData = metadata_ordered,
                              design = ~temperature*parent)
```

First we are going to log-transform the data using a variance stabilizing transforamtion (VST). This is only for visualization purposes. Essentially, this is roughly similar to putting the data on the log2 scale. It will deal with the sampling variability of low counts by calculating within-group variability (if blind=FALSE). Importantly, it does not use the design to remove variation in the data, and so can be used to examine if there may be any variability do to technical factors such as extraction batch effects.

To do this we first need to calculate the size factors of our samples. This is a rough estimate of how many reads each sample contains compared to the others. In order to use VST (the faster log2 transforming process) to log-transform our data, the size factors need to be less than 4.

Chunk should return TRUE if <4.  
```{r}
SF.gdds <- estimateSizeFactors(gdds) #estimate size factors to determine if we can use vst  to transform our data. Size factors should be less than 4 for us to use vst
print(sizeFactors(SF.gdds)) #View size factors

all(sizeFactors(SF.gdds)) < 4
```

All size factors are less than 4, so we can use VST transformation.  

```{r}
gvst <- vst(gdds, blind=FALSE) #apply a variance stabilizing transforamtion to minimize effects of small counts and normalize wrt library size
head(assay(gvst), 3) #view transformed gene count data for the first three genes in the dataset.  
```

# **Conduct PERMANOVA by parent and temperature**  

Export data for PERMANOVA test.  
```{r}
test<-t(assay(gvst)) #export as matrix
test<-as.data.frame(test)

#add category columns
test$sample<-rownames(test)
test$temperature<-metadata$temperature[match(test$sample, metadata$sample)]
test$parent<-metadata$parent[match(test$sample, metadata$sample)]
test$code<-metadata$code[match(test$sample, metadata$sample)]

```

Build PERMANOVA model for all genes in the dataset.  
```{r}
library(vegan)
library(factoextra)

scaled_test <-prcomp(test[c(1:24003)], scale=TRUE, center=TRUE)
fviz_eig(scaled_test)

# scale data
vegan <- scale(test[c(1:24003)])

# PerMANOVA 
permanova<-adonis2(vegan ~ parent*temperature, data = test, method='eu')
permanova
```

Permutation test for adonis under reduced model
Terms added sequentially (first to last)
Permutation: free
Number of permutations: 999

adonis2(formula = vegan ~ parent * temperature, data = test, method = "eu")
                   Df SumOfSqs      R2      F Pr(>F)    
parent              2   154649 0.12156 4.4308  0.001 ***
temperature         2   253981 0.19965 7.2767  0.001 ***
parent:temperature  4    78208 0.06148 1.1204  0.182    
Residual           45   785321 0.61731                  
Total              53  1272159 1.00000 

Gene expression is significantly different between temperature and parent, no interactive effect detected.  

# **Examine PCA and sample distances of all genes**  

Plot a PCA of samples by temperature and parental phenotype for all genes.        

```{r}
gPCAdata <- plotPCA(gvst, intgroup = c("temperature", "parent"), returnData=TRUE)
percentVar <- round(100*attr(gPCAdata, "percentVar")) #plot PCA of samples with all data

allgenesfilt_PCA <- ggplot(gPCAdata, aes(PC1, PC2, color=parent, shape=temperature)) + 
  geom_point(size=3) + 
  ggtitle(expression(bold("All genes (24,003 genes)")))+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  scale_color_manual(name="Parental Phenotype", values=c("orange", "brown4", "gray"))+
  scale_shape_manual(name="Temperature", values=c(19,17,15))+
  geom_text(x=0, y=10, label="p[Parent]=0.001", color="black")+
  geom_text(x=0, y=8.5, label="p[Temperature]=0.001", color="black")+
  geom_text(x=0, y=7, label="p[Interaction]=0.182", color="darkgray")+
  theme_classic() + #Set background color
  theme(panel.border = element_blank(), # Set border
                     #panel.grid.major = element_blank(), #Set major gridlines 
                     #panel.grid.minor = element_blank(), #Set minor gridlines
                     axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()); allgenesfilt_PCA

ggsave("figures/rna_seq/pca.pdf", allgenesfilt_PCA, width=8, height=4)
ggsave("figures/rna_seq/pca.jpeg", allgenesfilt_PCA, width=8, height=4)
```

Add a plot with elipses.  

```{r}
allgenes_ellipse<-allgenesfilt_PCA + stat_ellipse();allgenes_ellipse

ggsave("figures/rna_seq/pca_ellipse.pdf", allgenes_ellipse, width=9, height=5)
ggsave("figures/rna_seq/pca_ellipse.jpeg", allgenes_ellipse, width=9, height=5)
```

# **Run DEG analysis of DEGs for all treatments**  

Run differential expression test using a Wald model. 
```{r, message = FALSE}
DEG <- DESeq(gdds) #run differential expression test by temperature x parent using the Wald model

DEG.results.all <- results(DEG)
DEG.df<-as.data.frame(DEG.results.all)

DEG.results.all <- as.data.frame(subset(DEG.results.all, padj<0.01))
DEG.results.all <- as.data.frame(subset(DEG.results.all, abs(log2FoldChange)>0.5))
dim(DEG.results.all)

DEGlist <- gdds[rownames(DEG.results.all)]

DEGvst <- varianceStabilizingTransformation(DEGlist) #apply a variance stabilizing transforamtion to minimize effects of small counts and normalize
```

There are 83 genes differentially expressed at p<0.01 and log2FoldChange>0.5. We can return to these filtering thresholds later. 

View volcano plot. 
```{r}
#BiocManager::install('apeglm')
#BiocManager::install('EnhancedVolcano')

library(apeglm)
library(EnhancedVolcano)

res <- results(DEG)

volcano_plot<-EnhancedVolcano(res,
    lab = rownames(res),
    x = 'log2FoldChange',
    y = 'pvalue',
    title = '',
    pCutoff = 0.01,
    FCcutoff = 0.5,
    pointSize = 3.0,
    labSize = 0.0);volcano_plot

ggsave("figures/rna_seq/volcano.pdf", plot=volcano_plot, width=14, height=8)
ggsave("figures/rna_seq/volcano.png", plot=volcano_plot, width=14, height=8)
```

# **Add heatmap by parent and treatment**  

```{r}
df <- as.data.frame(colData(gdds)[,c("temperature","parent")])
df$temperature <- factor(as.character(df$temperature), levels=c("27", "30", "33"))
df$parent <- factor(as.character(df$parent), levels=c("Bleached", "Nonbleached", "Wildtype"))
df<-df%>%
  arrange(temperature, parent)

library(viridis); library(pheatmap)

ann_colors = list(
    temperature = c("27" = "lightblue", "30" = "pink2","33"= "red2"), 
    parent = c("Bleached" = "orange", "Nonbleached" = "brown4", "Wildtype" = "gray")
)

legend_names_col = colnames(assay(DEGvst))

pdf("figures/rna_seq/DEG_heatmap.pdf", width=10, height=10)
pheatmap(assay(DEGvst), cluster_rows=TRUE, show_rownames=FALSE, color=inferno(10), show_colnames=TRUE, fontsize_row=3, scale="row", cluster_cols=TRUE, annotation_col=df, annotation_colors = ann_colors, labels_col=legend_names_col)
dev.off()

```

# **Plot a heatmap of sample to sample distances**    

```{r}
gsampleDists <- dist(t(assay(gvst))) #calculate distance matix
gsampleDistMatrix <- as.matrix(gsampleDists) #distance matrix
rownames(gsampleDistMatrix) <- colnames(gvst) #assign row names
colnames(gsampleDistMatrix) <- NULL #assign col names
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255) #assign colors

save_pheatmap_pdf <- function(x, filename, width=7, height=7) {
   stopifnot(!missing(x))
   stopifnot(!missing(filename))
   pdf(filename, width=width, height=height)
   grid::grid.newpage()
   grid::grid.draw(x$gtable)
   dev.off()
}

pht<-pheatmap(gsampleDistMatrix, #plot matrix
         clustering_distance_rows=gsampleDists, #cluster rows
         clustering_distance_cols=gsampleDists, #cluster columns
         col=colors) #set colors

save_pheatmap_pdf(pht, "figures/rna_seq/sample_distances.pdf")
dev.off()

```

# DEG Analyses 

## Parent comparisons 

Specifically look for genes that are different between bleached and non bleached parental histories.  

### Nonbleached vs Bleached
```{r}
parent_degNB_B<-results(DEG, contrast=c("parent","Nonbleached","Bleached"))
parent_degNB_B <- as.data.frame(subset(parent_degNB_B, padj<0.01))
parent_degNB_B <- as.data.frame(subset(parent_degNB_B, abs(log2FoldChange)>0.5))

dim(parent_degNB_B)
```
There are 569 genes differentially expressed between nonbleached and bleached parents at p<0.01 and l2FC>0.5. 

```{r}
DEGlistNB_B <- gdds[rownames(parent_degNB_B)]
DEGvstNB_B <- varianceStabilizingTransformation(DEGlistNB_B) 

pdf("figures/rna_seq/DEG_heatmap_NBvsB.pdf", width=10, height=10)
pheatmap(assay(DEGvstNB_B), cluster_rows=TRUE, show_rownames=FALSE, color=inferno(10), show_colnames=TRUE, fontsize_row=3, scale="row", cluster_cols=TRUE, annotation_col=df, annotation_colors = ann_colors, labels_col=legend_names_col)
dev.off()
```


### Nonbleached vs Wildtype
```{r}
parent_parent_degNB_WT<-results(DEG, contrast=c("parent","Nonbleached","Wildtype"))
parent_parent_degNB_WT <- as.data.frame(subset(parent_parent_degNB_WT, padj<0.01))
parent_parent_degNB_WT <- as.data.frame(subset(parent_parent_degNB_WT, abs(log2FoldChange)>0.5))

dim(parent_parent_degNB_WT)
```
There are 526 genes differentially expressed between nonbleached and bleached parents at p<0.01 and l2FC>0.5. 

### Bleached vs Wildtype
```{r}
parent_parent_degB_WT<-results(DEG, contrast=c("parent","Bleached","Wildtype"))
parent_parent_degB_WT <- as.data.frame(subset(parent_parent_degB_WT, padj<0.01))
parent_parent_degB_WT <- as.data.frame(subset(parent_parent_degB_WT, abs(log2FoldChange)>0.5))

dim(parent_parent_degB_WT)
```
There are 546 genes differentially expressed between nonbleached and bleached parents at p<0.01 and l2FC>0.5. 

## Temperature comparisons 

Next, look at 27 vs 30 degrees and 27 vs 33 degrees. 

27 vs 30 degrees 
```{r}
temp30_deg<-results(DEG, contrast=c("temperature","30","27"))
temp30_deg <- as.data.frame(subset(temp30_deg, padj<0.01))
temp30_deg <- as.data.frame(subset(temp30_deg, abs(log2FoldChange)>0.5))

dim(temp30_deg)
```
There are 317 DEGs between 27 and 30 degrees. 

27 vs 33 degrees 
```{r}
temp33_deg<-results(DEG, contrast=c("temperature","33","27"))
temp33_deg <- as.data.frame(subset(temp33_deg, padj<0.01))
temp33_deg <- as.data.frame(subset(temp33_deg, abs(log2FoldChange)>0.5))

dim(temp33_deg)
```
There are 5009 DEGs between 27 and 33 degrees. 

```{r}
DEGlist27_33 <- gdds[rownames(temp33_deg)]
DEGvst27_33 <- varianceStabilizingTransformation(DEGlist27_33) 

pdf("figures/rna_seq/DEG_heatmap_27vs33.pdf", width=10, height=20)
pheatmap(assay(DEGvst27_33), cluster_rows=TRUE, show_rownames=FALSE, color=inferno(10), show_colnames=TRUE, fontsize_row=3, scale="row", cluster_cols=TRUE, annotation_col=df, annotation_colors = ann_colors, labels_col=legend_names_col)
dev.off()
```

